<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>geneset.enrichment — geneset.enrichment • DEprot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="geneset.enrichment — geneset.enrichment"><meta property="og:description" content="Perform Gene Set Enrichment Analyses (GSEA) or OverRepresentation Analyses (ORA) on a proteomics differential analyses."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEprot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/DEprot.overview.vignette.html">*DEprot*: R-package for LFQ proteomics data normalization, imputation and Differential Expression analyses</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/sebastian-gregoricchio/DEprot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>geneset.enrichment</h1>
    <small class="dont-index">Source: <a href="https://github.com/sebastian-gregoricchio/DEprot/blob/main/R/geneset.enrichment.R" class="external-link"><code>R/geneset.enrichment.R</code></a></small>
    <div class="hidden name"><code>geneset.enrichment.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform Gene Set Enrichment Analyses (GSEA) or OverRepresentation Analyses (ORA) on a proteomics differential analyses.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">geneset.enrichment</span><span class="op">(</span></span>
<span>  <span class="va">DEprot.analyses.object</span>,</span>
<span>  <span class="va">contrast</span>,</span>
<span>  <span class="va">TERM2GENE</span>,</span>
<span>  <span class="va">enrichment.type</span>,</span>
<span>  gsea.rank.method <span class="op">=</span> <span class="st">"foldchange"</span>,</span>
<span>  diff.status.category <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gsub.pattern.prot.id <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>  dotplot.n <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-deprot-analyses-object">DEprot.analyses.object<a class="anchor" aria-label="anchor" href="#arg-deprot-analyses-object"></a></dt>
<dd><p>An object of class <code>DEprot.analyses</code>.</p></dd>


<dt id="arg-contrast">contrast<a class="anchor" aria-label="anchor" href="#arg-contrast"></a></dt>
<dd><p>Number indicating the position of the contrast to use for the plotting.</p></dd>


<dt id="arg-term-gene">TERM2GENE<a class="anchor" aria-label="anchor" href="#arg-term-gene"></a></dt>
<dd><p>Data.frame containing two columns 'gs_name' (IDs of the gene sets) and 'gene_symbol' (indicating the gene IDs). No default.</p></dd>


<dt id="arg-enrichment-type">enrichment.type<a class="anchor" aria-label="anchor" href="#arg-enrichment-type"></a></dt>
<dd><p>String indicating the type of analyses to perform. One among: GSEA, ORA. No default.</p></dd>


<dt id="arg-gsea-rank-method">gsea.rank.method<a class="anchor" aria-label="anchor" href="#arg-gsea-rank-method"></a></dt>
<dd><p>String indicating the type of gene ranking to use for GSEA analyses. Possible options: <code>"foldchange"</code> (log2FC value of the contrast), <code>"correlation"</code> (spearman's correlation coefficient of the imputed counts between the two groups in the contrast). Default: <code>"foldchange"</code>.</p></dd>


<dt id="arg-diff-status-category">diff.status.category<a class="anchor" aria-label="anchor" href="#arg-diff-status-category"></a></dt>
<dd><p>String indicating a diff.status among the ones present in the results table of the specific contrast. Used only one 'ORA' is performed. Default: <code>NULL</code>.</p></dd>


<dt id="arg-gsub-pattern-prot-id">gsub.pattern.prot.id<a class="anchor" aria-label="anchor" href="#arg-gsub-pattern-prot-id"></a></dt>
<dd><p>String indicating a pattern to be passed to gsub and to remove from the prot.id. Default: <code>NULL</code> (non changes in the IDs).</p></dd>


<dt id="arg-pvaluecutoff">pvalueCutoff<a class="anchor" aria-label="anchor" href="#arg-pvaluecutoff"></a></dt>
<dd><p>Numeric value indicating the adjusted pvalue cutoff on enrichment tests to report. Default: <code>0.05</code>.</p></dd>


<dt id="arg-qvaluecutoff">qvalueCutoff<a class="anchor" aria-label="anchor" href="#arg-qvaluecutoff"></a></dt>
<dd><p>Numeric value indicating the qvalue cutoff on enrichment tests to report as significant (only for ORA). Tests must pass i) pvalueCutoff on unadjusted pvalues, ii) pvalueCutoff on adjusted pvalues and iii) qvalueCutoff on qvalues to be reported. Default: <code>0.05</code>.</p></dd>


<dt id="arg-padjustmethod">pAdjustMethod<a class="anchor" aria-label="anchor" href="#arg-padjustmethod"></a></dt>
<dd><p>String indicating the method to use for the p-value adjustment. One mong "holm", "hochberg", "hommel", "bonferroni", "BH", "BY", "fdr", "none". Default: <code>"BH"</code>.</p></dd>


<dt id="arg-dotplot-n">dotplot.n<a class="anchor" aria-label="anchor" href="#arg-dotplot-n"></a></dt>
<dd><p>Numeric value indicating the maximum number of categories to plot in the dotplot. Default: <code>10</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>An object of class <code>DEprot.enrichResult</code>.</p>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Sebastian Gregoricchio</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Perform Over-Representation Analyses (ORA)</span></span></span>
<span class="r-in"><span><span class="va">ora.results</span> <span class="op">&lt;-</span> <span class="fu">geneset.enrichment</span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="fu">DEprot</span><span class="fu">::</span><span class="va"><a href="test.toolbox.html">test.toolbox</a></span><span class="op">$</span><span class="va">diff.exp.limma</span>,</span></span>
<span class="r-in"><span>                                  contrast <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                                  TERM2GENE <span class="op">=</span> <span class="fu">DEprot</span><span class="fu">::</span><span class="va"><a href="test.toolbox.html">test.toolbox</a></span><span class="op">$</span><span class="va">geneset</span>,</span></span>
<span class="r-in"><span>                                  enrichment.type <span class="op">=</span> <span class="st">"ORA"</span>,</span></span>
<span class="r-in"><span>                                  diff.status.category <span class="op">=</span> <span class="st">"FBS"</span>,</span></span>
<span class="r-in"><span>                                  pvalueCutoff <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                                  qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Scale for <span style="color: #00BB00;">fill</span> is already present.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Please use `linewidth` instead.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> The deprecated feature was likely used in the <span style="color: #0000BB;">ggtangle</span> package.</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span>   Please report the issue to the authors.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Scale for <span style="color: #00BB00;">fill</span> is already present.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Perform GeneSet Enrichment Analyses (GSEA)</span></span></span>
<span class="r-in"><span><span class="va">gsea.results</span> <span class="op">&lt;-</span> <span class="fu">geneset.enrichment</span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="fu">DEprot</span><span class="fu">::</span><span class="va"><a href="test.toolbox.html">test.toolbox</a></span><span class="op">$</span><span class="va">diff.exp.limma</span>,</span></span>
<span class="r-in"><span>                                   contrast <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                                   TERM2GENE <span class="op">=</span> <span class="fu">DEprot</span><span class="fu">::</span><span class="va"><a href="test.toolbox.html">test.toolbox</a></span><span class="op">$</span><span class="va">geneset</span>,</span></span>
<span class="r-in"><span>                                   enrichment.type <span class="op">=</span> <span class="st">"GSEA"</span>,</span></span>
<span class="r-in"><span>                                   gsea.rank.method <span class="op">=</span> <span class="st">"foldchange"</span>,</span></span>
<span class="r-in"><span>                                   pvalueCutoff <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                                   qvalueCutoff <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> using 'fgsea' for GSEA analysis, please cite Korotkevich et al (2019).</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> preparing geneSet collections...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> GSEA analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> leading edge analysis...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> done...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Scale for <span style="color: #00BB00;">fill</span> is already present.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Adding another scale for <span style="color: #00BB00;">fill</span>, which will replace the existing scale.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Sebastian Gregoricchio.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

