<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>impute.counts — impute.counts • DEprot</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="impute.counts — impute.counts"><meta property="og:description" content="Function that allows for the imputation of missing values using 4 possible algorithms: missForest, kNN, LLS, SVD."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEprot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/DEprot.overview.vignette.html">*DEprot*: R-package for LFQ proteomics data normalization, imputation and Differential Expression analyses</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/sebastian-gregoricchio/DEprot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>impute.counts</h1>
    <small class="dont-index">Source: <a href="https://github.com/sebastian-gregoricchio/DEprot/blob/main/R/impute.counts.R" class="external-link"><code>R/impute.counts.R</code></a></small>
    <div class="hidden name"><code>impute.counts.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Function that allows for the imputation of missing values using 4 possible algorithms: <a href="https://www.rdocumentation.org/packages/missForest/" class="external-link">missForest</a>, <a href="https://rdrr.io/cran/VIM/man/kNN.html" class="external-link">kNN</a>, <a href="https://rdrr.io/bioc/pcaMethods/man/llsImpute.html" class="external-link">LLS</a>, <a href="https://rdrr.io/bioc/pcaMethods/man/pca.html" class="external-link">SVD</a>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">impute.counts</span><span class="op">(</span></span>
<span>  <span class="va">DEprot.object</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"missForest"</span>,</span>
<span>  use.normalized.data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  overwrite.imputation <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  missForest.max.iterations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  missForest.variable.wise.OOBerror <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  missForest.cores <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  missForest.parallel.mode <span class="op">=</span> <span class="st">"variables"</span>,</span>
<span>  kNN.n.nearest.neighbours <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  LLS.k <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  pcaMethods.nPCs.to.test <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  RegImpute.max.iterations <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  RegImpute.fillmethod <span class="op">=</span> <span class="st">"row_mean"</span>,</span>
<span>  seed <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-deprot-object">DEprot.object<a class="anchor" aria-label="anchor" href="#arg-deprot-object"></a></dt>
<dd><p>A <code>DEprot object</code>, as generated by <a href="load.counts.html">load.counts</a>.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>String indicating the imputation method to use. One among: 'missForest', 'kNN' (VIM), 'tkNN' (imputomics), 'corkNN' (imputomics), 'LLS' (pcaMethods), 'SVD' (a.k.a svdImpute, pcaMethods), 'BPCA' (pcaMethods), 'PPCA' (pcaMethods), 'RegImpute' (DreamAI). Default: <code>"missForest"</code>.</p></dd>


<dt id="arg-use-normalized-data">use.normalized.data<a class="anchor" aria-label="anchor" href="#arg-use-normalized-data"></a></dt>
<dd><p>Logical value indicating whether the imputation should be performed based on the rationalized data. Default: <code>TRUE</code>.</p></dd>


<dt id="arg-overwrite-imputation">overwrite.imputation<a class="anchor" aria-label="anchor" href="#arg-overwrite-imputation"></a></dt>
<dd><p>Logical value to indicate whether, in the case already available, the table of imputed counts should be overwritten. Default: <code>FALSE</code>.</p></dd>


<dt id="arg-missforest-max-iterations">missForest.max.iterations<a class="anchor" aria-label="anchor" href="#arg-missforest-max-iterations"></a></dt>
<dd><p>Max number of iterations for the missForest algorithm. Default: <code>100</code>.</p></dd>


<dt id="arg-missforest-variable-wise-ooberror">missForest.variable.wise.OOBerror<a class="anchor" aria-label="anchor" href="#arg-missforest-variable-wise-ooberror"></a></dt>
<dd><p>Logical value to define whether the OOB error is returned for each variable separately. Default: <code>TRUE</code>.</p></dd>


<dt id="arg-missforest-cores">missForest.cores<a class="anchor" aria-label="anchor" href="#arg-missforest-cores"></a></dt>
<dd><p>Number of cores used to run the <code>missForest</code> algorithm. If <code>missForest.cores</code> is 1 (or lower), the imputation will be run in parallel. Two modes are possible and can be defined by the parameter <code>missForest.parallel.mode</code>. Default: <code>1</code>.</p></dd>


<dt id="arg-missforest-parallel-mode">missForest.parallel.mode<a class="anchor" aria-label="anchor" href="#arg-missforest-parallel-mode"></a></dt>
<dd><p>Define the mode to use for the parallelization, ignored when <code>cores</code> is more than 1. One among: 'variables', 'forests'. Default: <code>"variables"</code>. See also the documentation of the <a href="https://www.rdocumentation.org/packages/missForest/versions/1.5/topics/missForest" class="external-link">missForest function</a>.</p></dd>


<dt id="arg-knn-n-nearest-neighbours">kNN.n.nearest.neighbours<a class="anchor" aria-label="anchor" href="#arg-knn-n-nearest-neighbours"></a></dt>
<dd><p>Numeric value indicating the number of nearest neighbors to use to perform the <code>kNN</code> imputation. Default: <code>10</code>.</p></dd>


<dt id="arg-lls-k">LLS.k<a class="anchor" aria-label="anchor" href="#arg-lls-k"></a></dt>
<dd><p>Cluster size, this is the number of similar genes used for regression. Default: <code>2</code>.</p></dd>


<dt id="arg-pcamethods-npcs-to-test">pcaMethods.nPCs.to.test<a class="anchor" aria-label="anchor" href="#arg-pcamethods-npcs-to-test"></a></dt>
<dd><p>Numeric value indicating the number of Principal Components to test in order to find the optimal number of PCs to used in the imputation methods from the <code>pcaMethods</code> package. This includes: 'LLS', 'SVD' (a.k.a 'svdImpute'), 'BPCA-pcaMethods', and 'PPCA'. Default: <code>5</code>.</p></dd>


<dt id="arg-regimpute-max-iterations">RegImpute.max.iterations<a class="anchor" aria-label="anchor" href="#arg-regimpute-max-iterations"></a></dt>
<dd><p>Numeric value indicating the number of maximum iteration for the imputation with <code>RegImpute</code> (from <code>DreamAI</code>). Default: <code>10</code>.</p></dd>


<dt id="arg-regimpute-fillmethod">RegImpute.fillmethod<a class="anchor" aria-label="anchor" href="#arg-regimpute-fillmethod"></a></dt>
<dd><p>String identifying the fill method to be used in the <code>RegImpute</code> method (from<code>DreamAI</code>). One among <code>"row_mean"</code> and <code>"zeros"</code>. Default: <code>"row_mean"</code>. It throws an warning if <code>"row_median"</code> is used.</p></dd>


<dt id="arg-seed">seed<a class="anchor" aria-label="anchor" href="#arg-seed"></a></dt>
<dd><p>Numeric value indicating the seed to use for the randomization. Default: <code>NULL</code>, automatically generated (saved in the <code>seed</code> element in the final imputation method list).</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical valued indicating whether processing messages should be printed. Default: <code>FALSE</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>A <code>DEprot</code> object. The boxplot showing the distribution of the protein intensity is remade and added to the slot (<code>boxplot.imputed</code>). A list with parameters and other info about the imputation is added as well in the <code>imputation</code> slot.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="https://www.rdocumentation.org/packages/missForest/" class="external-link">missForest</a>, <a href="https://cran.r-project.org/web/packages/VIM/index.html" class="external-link">VIM</a>, <a href="https://www.bioconductor.org/packages/release/bioc/html/pcaMethods.html" class="external-link">pcaMethods</a> R-packages, <a href="https://github.com/sebastian-gregoricchio/DreamAI/" class="external-link">DreamAI</a>, <a href="https://github.com/BioGenies/imputomics" class="external-link">imputomics</a>.</p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Sebastian Gregoricchio</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">dpo</span> <span class="op">&lt;-</span> <span class="fu">impute.counts</span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="fu">DEprot</span><span class="fu">::</span><span class="va"><a href="test.toolbox.html">test.toolbox</a></span><span class="op">$</span><span class="va">dpo.norm</span>,</span></span>
<span class="r-in"><span>                     method <span class="op">=</span> <span class="st">"missForest"</span>,</span></span>
<span class="r-in"><span>                     missForest.cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>Foreach loop (doParallelMC) had changed the current RNG type: RNG was restored to same type, next state</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span>The response has five or fewer unique values.  Are you sure you want to do regression?</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Sebastian Gregoricchio.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer></div>






  </body></html>

