<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>*DEprot*: R-package for LFQ proteomics data normalization, imputation and Differential Expression analyses • DEprot</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="*DEprot*: R-package for LFQ proteomics data normalization, imputation and Differential Expression analyses">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">DEprot</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/DEprot.overview.vignette.html">*DEprot*: R-package for LFQ proteomics data normalization, imputation and Differential Expression analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/sebastian-gregoricchio/DEprot/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>
<em>DEprot</em>: R-package for LFQ proteomics
data normalization, imputation and Differential Expression analyses</h1>
                        <h4 data-toc-skip class="author">Sebastian
Gregoricchio, Ph.D.</h4>
            <address class="author_afil">
      Netherlands Cancer Institute, Oncogenomics division, Amsterdam -
Netherlands<br><a class="author_email" href="mailto:#"></a><a href="mailto:s.gregoricchio@nki.nl" class="email">s.gregoricchio@nki.nl</a>
      </address>
                              <h4 data-toc-skip class="author">Nils
Eickhoff, Ph.D.</h4>
            <address class="author_afil">
      Netherlands Cancer Institute, Oncogenomics division, Amsterdam -
Netherlands<br><a class="author_email" href="mailto:#"></a><a href="mailto:n.eickhoff@nki.nl" class="email">n.eickhoff@nki.nl</a>
      </address>
                  
            <h4 data-toc-skip class="date">Last update: 15<sup>th</sup>
August, 2025</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/sebastian-gregoricchio/DEprot/blob/main/vignettes/DEprot.overview.vignette.Rmd" class="external-link"><code>vignettes/DEprot.overview.vignette.Rmd</code></a></small>
      <div class="hidden name"><code>DEprot.overview.vignette.Rmd</code></div>

    </div>

    
    
<style>
body {
text-align: justify}
</style>
<p><img src="https://img.shields.io/github/v/release/sebastian-gregoricchio/DEprot" alt="release"><img src="https://img.shields.io/badge/License-GPLv3-blue.png" alt="license"></p>
<p><a href="https://sebastian-gregoricchio.github.io/DEprot" class="external-link"><img src="https://sebastian-gregoricchio.github.io/DEprot/DEprot_logo.png" align="right" height="150/"></a></p>
<hr>
<div class="section level2">
<h2 id="intro">
<strong>Introduction</strong><a class="anchor" aria-label="anchor" href="#intro"></a>
</h2>
<p>The concept behind the <em>DEprot</em> (<em>D</em>ifferential
<em>E</em>xpression <em>prot</em>eomics) is to provide a toolkit that
allows for the normalization, imputation and analyses of the
differential protein expression in proteomics data. The data are assumed
to be LFQ (label-free quantitation) values.</p>
<p><br></p>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>If you use this package, please cite:</p>
<div class="warning" style="padding:2.5%; background-color:#c1eac4; color:#226a27; margin-left:5%; margin-right:5%; border-radius:15px;">
<span> <font size="-0.5">
<p style="margin-top:1em; text-align:left">
<strong>Citation</strong>
</p>
<div style="margin-left:2%; margin-right:2%; text-align: justify">
<p>“No publication associated yet <em>et al.</em>, <em>XYZ</em> (2123).
<br><em>doi</em>: <a href="https://doi.org/XYZ" class="external-link">XYZ/XYZ</a></p>
</div>
<p></p></font>
<p></p></span>
</div>
<p><br></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">citation</span>(<span class="st">"DEprot"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="sc">&gt;</span> To cite package <span class="st">'DEprot'</span> <span class="cf">in</span> publications use<span class="sc">:</span></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="er">&gt;</span> </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="er">&gt;</span>   Gregoricchio <span class="fu">S</span> (<span class="dv">2025</span>). _DEprot<span class="sc">:</span> An R<span class="sc">-</span>package <span class="cf">for</span> proteomics</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="sc">&gt;</span>   differential analyses_. R package version <span class="dv">0</span>.<span class="fl">1.0</span>,</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="sc">&gt;</span>   <span class="er">&lt;</span>https<span class="sc">:</span><span class="er">//</span>sebastian<span class="sc">-</span>gregoricchio.github.io<span class="sc">/</span>DEprot<span class="sc">/</span><span class="er">&gt;</span>.</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="sc">&gt;</span> </span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="er">&gt;</span> A BibTeX entry <span class="cf">for</span> LaTeX users is</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="sc">&gt;</span> </span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="er">&gt;</span>   <span class="er">@</span>Manual{,</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="sc">&gt;</span>     title <span class="ot">=</span> {DEprot<span class="sc">:</span> An R<span class="sc">-</span>package <span class="cf">for</span> proteomics differential analyses},</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="sc">&gt;</span>     author <span class="ot">=</span> {Sebastian Gregoricchio},</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="sc">&gt;</span>     year <span class="ot">=</span> {<span class="dv">2025</span>},</span>
<span id="cb1-14"><a href="#cb1-14" tabindex="-1"></a><span class="sc">&gt;</span>     note <span class="ot">=</span> {R package version <span class="dv">0</span>.<span class="fl">1.0</span>},</span>
<span id="cb1-15"><a href="#cb1-15" tabindex="-1"></a><span class="sc">&gt;</span>     url <span class="ot">=</span> {https<span class="sc">:</span><span class="er">//</span>sebastian<span class="sc">-</span>gregoricchio.github.io<span class="sc">/</span>DEprot<span class="sc">/</span>},</span>
<span id="cb1-16"><a href="#cb1-16" tabindex="-1"></a><span class="sc">&gt;</span>   }</span>
<span id="cb1-17"><a href="#cb1-17" tabindex="-1"></a><span class="sc">&gt;</span> </span>
<span id="cb1-18"><a href="#cb1-18" tabindex="-1"></a><span class="er">&gt;</span> ATTENTION<span class="sc">:</span> This citation information has been auto<span class="sc">-</span>generated from the</span>
<span id="cb1-19"><a href="#cb1-19" tabindex="-1"></a><span class="sc">&gt;</span> package DESCRIPTION file and may need manual editing, see</span>
<span id="cb1-20"><a href="#cb1-20" tabindex="-1"></a><span class="sc">&gt;</span> <span class="st">'help("citation")'</span>.</span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="general-workflow">General workflow<a class="anchor" aria-label="anchor" href="#general-workflow"></a>
</h3>
<p>Data can be loaded as raw values, pre-normalized unimputed LFQ values
or, imputed LFQ values. Further, if multiple batches are available, it
is possible to perform batch corrections by combining LFQ uimputed
tables and load this data as “raw” in order to perform a batch
harmonization.</p>
<p>Using normalized and imputed data, <code>DEprot</code> will help you
in the differential analyses as well in performing quality controls on
your data (sample correlation, principal components analyses (PCA)).</p>
<p><br></p>
<p><img src="https://sebastian-gregoricchio.github.io/DEprot/resources/DEprot_workflow.png"></p>
<p><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="loadingdata">
<strong>Loading the data</strong><a class="anchor" aria-label="anchor" href="#loadingdata"></a>
</h2>
<p>The package starting point is the building of a <code>DEprot</code>
object. This class of objects is specific to this package and requires
at least two elements:</p>
<ul>
<li>
<em>counts table</em>: this must be a matrix in which the column
names represent the samples while the row names identify the proteins.
The values in the matrix are assumed to be LFQ (label-free
quantification) values, either in linear or log-transformed format.</li>
<li>
<em>metadata</em>: this must be a data.frame containing at least one
column names <code>column.id</code> which values correspond to the
column names of the <em>counts_table</em>. Any other additional column
can be added (cell lines, treatment, condition, timing, etc) and used to
define groups for differential and quality control (<a href="#PCA">PCA</a> and <a href="#correlation">correlation</a>)
analyses.</li>
</ul>
<p>In the next paragraph we will refer to a dataset (pre-loaded in
<code>DEprot</code>), in which a breast cancer (BCa) cell line was
cultured in either hormone-deprived media or in full media (FBS). If
cultured in hormone-deprived condition, it was then treated for 6 hrs
with either
<math display="inline" xmlns="http://www.w3.org/1998/Math/MathML"><semantics><mi>β</mi><annotation encoding="application/x-tex">\beta</annotation></semantics></math>-estradiol
(E2) or vehicle (DMSO). 4 biological replicates have been analyzed.
Hence, the datasets consists of 1 cell lines x 3 conditions x 4
replicates, for a total of 12 sample. Proteins and samples have been
“anonymized”.</p>
<p><br></p>
<div class="section level3">
<h3 id="preloadeddata">Collect pre-loaded data<a class="anchor" aria-label="anchor" href="#preloadeddata"></a>
</h3>
<p>The counts represent LFQ values in the log2 format and are
not-imputed.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Metadata</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"sample.config"</span>, package <span class="op">=</span> <span class="st">"DEprot"</span><span class="op">)</span></span>
<span><span class="va">sample.config</span></span></code></pre></div>
<table class="table">
<caption><strong>Sample metadata table</strong></caption>
<colgroup>
<col width="14%">
<col width="28%">
<col width="7%">
<col width="15%">
<col width="21%">
<col width="14%">
</colgroup>
<thead><tr class="header">
<th align="left">column.id</th>
<th align="left">sample.id</th>
<th align="left">cell</th>
<th align="left">condition</th>
<th align="left">combined.id</th>
<th align="left">replicate</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Sample_A</td>
<td align="left">BCa_FBS_rep1</td>
<td align="left">BCa</td>
<td align="left">FBS</td>
<td align="left">BCa_FBS</td>
<td align="left">rep1</td>
</tr>
<tr class="even">
<td align="left">Sample_B</td>
<td align="left">BCa_6h.DMSO_rep1</td>
<td align="left">BCa</td>
<td align="left">6h.DMSO</td>
<td align="left">BCa_6h.DMSO</td>
<td align="left">rep1</td>
</tr>
<tr class="odd">
<td align="left">Sample_C</td>
<td align="left">BCa_6h.10nM.E2_rep1</td>
<td align="left">BCa</td>
<td align="left">6h.10nM.E2</td>
<td align="left">BCa_6h.10nM.E2</td>
<td align="left">rep1</td>
</tr>
<tr class="even">
<td align="left">Sample_D</td>
<td align="left">BCa_FBS_rep2</td>
<td align="left">BCa</td>
<td align="left">FBS</td>
<td align="left">BCa_FBS</td>
<td align="left">rep2</td>
</tr>
<tr class="odd">
<td align="left">Sample_E</td>
<td align="left">BCa_6h.DMSO_rep2</td>
<td align="left">BCa</td>
<td align="left">6h.DMSO</td>
<td align="left">BCa_6h.DMSO</td>
<td align="left">rep2</td>
</tr>
<tr class="even">
<td align="left">Sample_F</td>
<td align="left">BCa_6h.10nM.E2_rep2</td>
<td align="left">BCa</td>
<td align="left">6h.10nM.E2</td>
<td align="left">BCa_6h.10nM.E2</td>
<td align="left">rep2</td>
</tr>
<tr class="odd">
<td align="left">Sample_G</td>
<td align="left">BCa_FBS_rep3</td>
<td align="left">BCa</td>
<td align="left">FBS</td>
<td align="left">BCa_FBS</td>
<td align="left">rep3</td>
</tr>
<tr class="even">
<td align="left">Sample_H</td>
<td align="left">BCa_6h.DMSO_rep3</td>
<td align="left">BCa</td>
<td align="left">6h.DMSO</td>
<td align="left">BCa_6h.DMSO</td>
<td align="left">rep3</td>
</tr>
<tr class="odd">
<td align="left">Sample_I</td>
<td align="left">BCa_6h.10nM.E2_rep3</td>
<td align="left">BCa</td>
<td align="left">6h.10nM.E2</td>
<td align="left">BCa_6h.10nM.E2</td>
<td align="left">rep3</td>
</tr>
<tr class="even">
<td align="left">Sample_J</td>
<td align="left">BCa_FBS_rep4</td>
<td align="left">BCa</td>
<td align="left">FBS</td>
<td align="left">BCa_FBS</td>
<td align="left">rep4</td>
</tr>
<tr class="odd">
<td align="left">Sample_K</td>
<td align="left">BCa_6h.DMSO_rep4</td>
<td align="left">BCa</td>
<td align="left">6h.DMSO</td>
<td align="left">BCa_6h.DMSO</td>
<td align="left">rep4</td>
</tr>
<tr class="even">
<td align="left">Sample_L</td>
<td align="left">BCa_6h.10nM.E2_rep4</td>
<td align="left">BCa</td>
<td align="left">6h.10nM.E2</td>
<td align="left">BCa_6h.10nM.E2</td>
<td align="left">rep4</td>
</tr>
</tbody>
</table>
<p><br></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># log2(LFQ) values (not imputed)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"unimputed.counts"</span>, package <span class="op">=</span> <span class="st">"DEprot"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">unimputed.counts</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption><strong>Unimputed log2(LFQ) values</strong></caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">Sample_A</th>
<th align="right">Sample_B</th>
<th align="right">Sample_C</th>
<th align="right">Sample_D</th>
<th align="right">Sample_E</th>
<th align="right">Sample_F</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">protein.1</td>
<td align="right">17.8663</td>
<td align="right">18.4581</td>
<td align="right">18.7380</td>
<td align="right">18.1438</td>
<td align="right">18.7811</td>
<td align="right">19.3043</td>
</tr>
<tr class="even">
<td align="left">protein.2</td>
<td align="right">22.5238</td>
<td align="right">22.9057</td>
<td align="right">23.0183</td>
<td align="right">22.3577</td>
<td align="right">23.0094</td>
<td align="right">23.1491</td>
</tr>
<tr class="odd">
<td align="left">protein.3</td>
<td align="right">16.1388</td>
<td align="right">15.3720</td>
<td align="right">15.2374</td>
<td align="right">17.0161</td>
<td align="right">15.5452</td>
<td align="right">15.7899</td>
</tr>
<tr class="even">
<td align="left">protein.4</td>
<td align="right">21.0189</td>
<td align="right">21.0513</td>
<td align="right">21.0460</td>
<td align="right">21.2229</td>
<td align="right">21.3352</td>
<td align="right">21.0178</td>
</tr>
<tr class="odd">
<td align="left">protein.5</td>
<td align="right">19.1038</td>
<td align="right">19.4950</td>
<td align="right">19.5529</td>
<td align="right">18.5697</td>
<td align="right">19.3690</td>
<td align="right">19.5156</td>
</tr>
<tr class="even">
<td align="left">protein.6</td>
<td align="right">15.8377</td>
<td align="right">15.4202</td>
<td align="right">15.4765</td>
<td align="right">15.9765</td>
<td align="right">15.3439</td>
<td align="right">15.2335</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div class="section level3">
<h3 id="deprotobject">Build <code>DEprot</code> object<a class="anchor" aria-label="anchor" href="#deprotobject"></a>
</h3>
<p>Now we will combine the the counts and metadata to create a
<code>DEprot</code> object (hereafter defined as <em>dpo</em>).<br>
Notice that another import information is whether the data are log
transformed, and if yes, which is the log base. Recommended
transformation is the <em>log<sub>2</sub>(score + 1)</em>.</p>
<p>If data are pre-normalized and/or pre-imputed, it can be indicated
with the corresponding parameters.<br> If this is not the case,
leave the parameters as <code>NA</code> (not <code>NULL</code>!).</p>
<p>Ultimately, if the metadata table does not contain a
<code>column.id</code> column corresponding to the row names of the
<code>counts_table</code>, it is possible to indicate the name of
another column that should be assumed as to be the
<code>column.id</code>.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a>dpo <span class="ot">&lt;-</span> <span class="fu">load.counts</span>(<span class="at">counts =</span> unimputed.counts,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>                   <span class="at">metadata =</span> sample.config,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>                   <span class="at">log.base =</span> <span class="dv">2</span>,</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>                   <span class="at">imputation =</span> <span class="cn">NA</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>                   <span class="at">normalization.method =</span> <span class="cn">NA</span>,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>                   <span class="at">column.id =</span> <span class="st">"column.id"</span>)</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>dpo</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a><span class="sc">&gt;</span> DEprot object<span class="sc">:</span></span>
<span id="cb4-10"><a href="#cb4-10" tabindex="-1"></a><span class="er">&gt;</span>            Samples<span class="sc">:</span>  <span class="dv">12</span></span>
<span id="cb4-11"><a href="#cb4-11" tabindex="-1"></a><span class="sc">&gt;</span>           Proteins<span class="sc">:</span>  <span class="dv">12944</span></span>
<span id="cb4-12"><a href="#cb4-12" tabindex="-1"></a><span class="sc">&gt;</span>   Counts available<span class="sc">:</span>  raw</span>
<span id="cb4-13"><a href="#cb4-13" tabindex="-1"></a><span class="sc">&gt;</span> Log transformation<span class="sc">:</span>  log2</span>
<span id="cb4-14"><a href="#cb4-14" tabindex="-1"></a><span class="sc">&gt;</span>   Metadata columns<span class="sc">:</span>  column.id, sample.id, cell, condition, combined.id, replicate</span></code></pre></div>
<p><br></p>
This object is an S4-vector of class <code>DEprot</code>. The S4-vectors
are containers of slots that can be accessed using the symbol
<code>@</code> (e.g., <code>object@slot.id</code>.<br>The structure
of an object of class <code>DEprot</code> (and
<code>DEprot.analyses</code>) is the following:
</div>
<table class="table">
<colgroup>
<col width="22%">
<col width="77%">
</colgroup>
<thead><tr class="header">
<th align="right">Slot</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><em>raw.counts</em></td>
<td align="left">table containing the raw counts, if not available it
will be <code>NULL</code>
</td>
</tr>
<tr class="even">
<td align="right"><em>norm.counts</em></td>
<td align="left">table containing the normalized counts, if not
available it will be <code>NULL</code>
</td>
</tr>
<tr class="odd">
<td align="right"><em>imputed.counts</em></td>
<td align="left">table containing the imputed counts, if not available
it will be <code>NULL</code>
</td>
</tr>
<tr class="even">
<td align="right"><em>log.base</em></td>
<td align="left">a number indicating the base of the log used to
transform the table, if not available it will be <code>NA</code>
</td>
</tr>
<tr class="odd">
<td align="right"><em>log.transformed</em></td>
<td align="left">logical value indicating whether the data are
log-transformed or not</td>
</tr>
<tr class="even">
<td align="right"><em>imputed</em></td>
<td align="left">logical value indicating whether the data are imputed
or not</td>
</tr>
<tr class="odd">
<td align="right"><em>imputation</em></td>
<td align="left">discussed further in the <a href="#imputation">Imputation paragraph</a>
</td>
</tr>
<tr class="even">
<td align="right"><em>normalized</em></td>
<td align="left">logical value indicating whether the data are
normalized or not</td>
</tr>
<tr class="odd">
<td align="right"><em>normalization.method</em></td>
<td align="left">a string indicating the type of normalization applied,
if not available it will be <code>NA</code>
</td>
</tr>
<tr class="even">
<td align="right"><em>boxplot.raw</em></td>
<td align="left">box+violin plot of the distribution of the LFQ
intensities per sample obtained from the raw counts</td>
</tr>
<tr class="odd">
<td align="right"><em>boxplot.norm</em></td>
<td align="left">box+violin plot of the distribution of the LFQ
intensities per sample obtained from the normalized counts</td>
</tr>
<tr class="even">
<td align="right"><em>boxplot.imputed</em></td>
<td align="left">box+violin plot of the distribution of the LFQ
intensities per sample obtained from the imputed counts</td>
</tr>
<tr class="odd">
<td align="right"><em>analyses.result.list</em></td>
<td align="left">discussed further in the <a href="#diffexp">Differential Expression analyses paragraph</a>
</td>
</tr>
<tr class="even">
<td align="right"><em>contrasts</em></td>
<td align="left">discussed further in the <a href="#diffexp">Differential Expression analyses paragraph</a>
</td>
</tr>
<tr class="odd">
<td align="right"><em>differential.analyses.params</em></td>
<td align="left">discussed further in the <a href="#diffexp">Differential Expression analyses paragraph</a>
</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div class="section level3">
<h3 id="rename">Rename sample columns<a class="anchor" aria-label="anchor" href="#rename"></a>
</h3>
<p>As in our example, sometimes the columns of the counts are not the
actual ID of the samples, but rather an identifier. However, it is
possible to rename the counts column names indicating any column of the
metadata table (having unique values). The original identifiers are
stored in a new column (<code>old.column.id</code>) of the
metadata.<br> Notice that the renaming will applied to all counts
table available.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>dpo <span class="ot">&lt;-</span> <span class="fu">rename.samples</span>(<span class="at">DEprot.object =</span> dpo,</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>                      <span class="at">metadata.column =</span> <span class="st">"sample.id"</span>)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a><span class="fu">get.metadata</span>(dpo)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a><span class="sc">&gt;</span>              column.id           sample.id cell  condition    combined.id</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">1</span>         BCa_FBS_rep1        BCa_FBS_rep1  BCa        FBS        BCa_FBS</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">2</span>     BCa_6h.DMSO_rep1    BCa_6h.DMSO_rep1  BCa    <span class="dv">6</span>h.DMSO    BCa_6h.DMSO</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">3</span>  BCa_6h<span class="fl">.10</span>nM.E2_rep1 BCa_6h<span class="fl">.10</span>nM.E2_rep1  BCa <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 BCa_6h<span class="fl">.10</span>nM.E2</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">4</span>         BCa_FBS_rep2        BCa_FBS_rep2  BCa        FBS        BCa_FBS</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">5</span>     BCa_6h.DMSO_rep2    BCa_6h.DMSO_rep2  BCa    <span class="dv">6</span>h.DMSO    BCa_6h.DMSO</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">6</span>  BCa_6h<span class="fl">.10</span>nM.E2_rep2 BCa_6h<span class="fl">.10</span>nM.E2_rep2  BCa <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 BCa_6h<span class="fl">.10</span>nM.E2</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">7</span>         BCa_FBS_rep3        BCa_FBS_rep3  BCa        FBS        BCa_FBS</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">8</span>     BCa_6h.DMSO_rep3    BCa_6h.DMSO_rep3  BCa    <span class="dv">6</span>h.DMSO    BCa_6h.DMSO</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">9</span>  BCa_6h<span class="fl">.10</span>nM.E2_rep3 BCa_6h<span class="fl">.10</span>nM.E2_rep3  BCa <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 BCa_6h<span class="fl">.10</span>nM.E2</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">10</span>        BCa_FBS_rep4        BCa_FBS_rep4  BCa        FBS        BCa_FBS</span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">11</span>    BCa_6h.DMSO_rep4    BCa_6h.DMSO_rep4  BCa    <span class="dv">6</span>h.DMSO    BCa_6h.DMSO</span>
<span id="cb5-17"><a href="#cb5-17" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">12</span> BCa_6h<span class="fl">.10</span>nM.E2_rep4 BCa_6h<span class="fl">.10</span>nM.E2_rep4  BCa <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 BCa_6h<span class="fl">.10</span>nM.E2</span>
<span id="cb5-18"><a href="#cb5-18" tabindex="-1"></a><span class="sc">&gt;</span>    replicate old.column.id</span>
<span id="cb5-19"><a href="#cb5-19" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">1</span>       rep1      Sample_A</span>
<span id="cb5-20"><a href="#cb5-20" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">2</span>       rep1      Sample_B</span>
<span id="cb5-21"><a href="#cb5-21" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">3</span>       rep1      Sample_C</span>
<span id="cb5-22"><a href="#cb5-22" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">4</span>       rep2      Sample_D</span>
<span id="cb5-23"><a href="#cb5-23" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">5</span>       rep2      Sample_E</span>
<span id="cb5-24"><a href="#cb5-24" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">6</span>       rep2      Sample_F</span>
<span id="cb5-25"><a href="#cb5-25" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">7</span>       rep3      Sample_G</span>
<span id="cb5-26"><a href="#cb5-26" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">8</span>       rep3      Sample_H</span>
<span id="cb5-27"><a href="#cb5-27" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">9</span>       rep3      Sample_I</span>
<span id="cb5-28"><a href="#cb5-28" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">10</span>      rep4      Sample_J</span>
<span id="cb5-29"><a href="#cb5-29" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">11</span>      rep4      Sample_K</span>
<span id="cb5-30"><a href="#cb5-30" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">12</span>      rep4      Sample_L</span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">head</span>(dpo<span class="sc">@</span>raw.counts[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a><span class="sc">&gt;</span>           BCa_FBS_rep1 BCa_6h.DMSO_rep1 BCa_6h<span class="fl">.10</span>nM.E2_rep1 BCa_FBS_rep2</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.1</span>      <span class="fl">17.8663</span>          <span class="fl">18.4581</span>             <span class="fl">18.7380</span>      <span class="fl">18.1438</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.2</span>      <span class="fl">22.5238</span>          <span class="fl">22.9057</span>             <span class="fl">23.0183</span>      <span class="fl">22.3577</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.3</span>      <span class="fl">16.1388</span>          <span class="fl">15.3720</span>             <span class="fl">15.2374</span>      <span class="fl">17.0161</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.4</span>      <span class="fl">21.0189</span>          <span class="fl">21.0513</span>             <span class="fl">21.0460</span>      <span class="fl">21.2229</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.5</span>      <span class="fl">19.1038</span>          <span class="fl">19.4950</span>             <span class="fl">19.5529</span>      <span class="fl">18.5697</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.6</span>      <span class="fl">15.8377</span>          <span class="fl">15.4202</span>             <span class="fl">15.4765</span>      <span class="fl">15.9765</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a><span class="sc">&gt;</span>           BCa_6h.DMSO_rep2 BCa_6h<span class="fl">.10</span>nM.E2_rep2</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.1</span>          <span class="fl">18.7811</span>             <span class="fl">19.3043</span></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.2</span>          <span class="fl">23.0094</span>             <span class="fl">23.1491</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.3</span>          <span class="fl">15.5452</span>             <span class="fl">15.7899</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.4</span>          <span class="fl">21.3352</span>             <span class="fl">21.0178</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.5</span>          <span class="fl">19.3690</span>             <span class="fl">19.5156</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.6</span>          <span class="fl">15.3439</span>             <span class="fl">15.2335</span></span></code></pre></div>
<p><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="normalization">
<strong>Data normalization</strong><a class="anchor" aria-label="anchor" href="#normalization"></a>
</h2>
<p>When a <code>DEprot</code> object is loaded, automatically a
box/violin plot showing the distribution of the LFQ values per samples
is generated.<br> This representation is useful to estimate whether
the data are normalized or not. Boxplots display the quantiles of the
LFQ intensities, while red and blue dashed lines correspond to maximum
and minimum LFQ value for each sample.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.raw</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/display_boxplot_raw-1.png" width="480" style="display: block; margin: auto;"></p>
<p>In this package we apply the Modified Balanced Quantile Normalization
(MBQN) from the <a href="https://www.bioconductor.org/packages/release/bioc/html/MBQN.html" class="external-link"><code>MBQN</code></a>
package and developed by <a href="https://doi.org/10.1002/pmic.202000068" class="external-link">E.Brombacher <em>et
al.</em> (<em>Proteomics</em>, 2020)</a>. The modification balances the
median (or mean) intensity of features (rows) which are rank invariant
(RI) or nearly rank invariant (NRI) across samples (columns) before
quantile normalization. This prevents an over-correction of the
intensity profiles of RI and NRI features by classical quantile
normalization and therefore supports the reduction of systematics in
downstream analyses.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>dpo <span class="ot">&lt;-</span> <span class="fu">normalize.counts</span>(<span class="at">DEprot.object =</span> dpo,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>                        <span class="at">NRI.RI.ratio.threshold =</span> <span class="fl">0.5</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>                        <span class="at">balancing.function =</span> <span class="st">"median"</span>)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a></span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>dpo</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a><span class="sc">&gt;</span> DEprot object<span class="sc">:</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a><span class="er">&gt;</span>            Samples<span class="sc">:</span>  <span class="dv">12</span></span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a><span class="sc">&gt;</span>           Proteins<span class="sc">:</span>  <span class="dv">12944</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="sc">&gt;</span>   Counts available<span class="sc">:</span>  raw, normalized</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a><span class="sc">&gt;</span> Log transformation<span class="sc">:</span>  log2</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a><span class="sc">&gt;</span>   Metadata columns<span class="sc">:</span>  column.id, sample.id, cell, condition, combined.id, replicate, old.column.id</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>dpo<span class="sc">@</span>normalization.method</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a><span class="sc">&gt;</span>                    param                  value</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">1</span>                package                   MBQN</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">2</span>                 method Quantile normalization</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">3</span>               balanced                   <span class="cn">TRUE</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">4</span>               <span class="cf">function</span>                 median</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">5</span> NRI<span class="sc">/</span>RI ratio threshold                    <span class="fl">0.5</span></span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a><span class="fu">head</span>(dpo<span class="sc">@</span>raw.counts[,<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>])</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a><span class="sc">&gt;</span>           BCa_FBS_rep1 BCa_6h.DMSO_rep1 BCa_6h<span class="fl">.10</span>nM.E2_rep1 BCa_FBS_rep2</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.1</span>      <span class="fl">17.8663</span>          <span class="fl">18.4581</span>             <span class="fl">18.7380</span>      <span class="fl">18.1438</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.2</span>      <span class="fl">22.5238</span>          <span class="fl">22.9057</span>             <span class="fl">23.0183</span>      <span class="fl">22.3577</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.3</span>      <span class="fl">16.1388</span>          <span class="fl">15.3720</span>             <span class="fl">15.2374</span>      <span class="fl">17.0161</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.4</span>      <span class="fl">21.0189</span>          <span class="fl">21.0513</span>             <span class="fl">21.0460</span>      <span class="fl">21.2229</span></span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.5</span>      <span class="fl">19.1038</span>          <span class="fl">19.4950</span>             <span class="fl">19.5529</span>      <span class="fl">18.5697</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.6</span>      <span class="fl">15.8377</span>          <span class="fl">15.4202</span>             <span class="fl">15.4765</span>      <span class="fl">15.9765</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a><span class="sc">&gt;</span>           BCa_6h.DMSO_rep2 BCa_6h<span class="fl">.10</span>nM.E2_rep2</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.1</span>          <span class="fl">18.7811</span>             <span class="fl">19.3043</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.2</span>          <span class="fl">23.0094</span>             <span class="fl">23.1491</span></span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.3</span>          <span class="fl">15.5452</span>             <span class="fl">15.7899</span></span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.4</span>          <span class="fl">21.3352</span>             <span class="fl">21.0178</span></span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.5</span>          <span class="fl">19.3690</span>             <span class="fl">19.5156</span></span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a><span class="sc">&gt;</span> protein<span class="fl">.6</span>          <span class="fl">15.3439</span>             <span class="fl">15.2335</span></span></code></pre></div>
<p><br></p>
<p>Also in this case a box/violin plot with the corresponding normalized
LFQ values per each sample is generated and stored in a vector slot.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.raw</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.norm</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/plot_norm-1.png" width="768" style="display: block; margin: auto;"></p>
<p><br></p>
<div class="section level3">
<h3 id="batch-effect-correction">Batch effect correction<a class="anchor" aria-label="anchor" href="#batch-effect-correction"></a>
</h3>
<p>Proteomics is highly susceptible to batch effects. Hereafter we use
the package <a href="https://www.bioconductor.org/packages/release/bioc/html/HarmonizR.html" class="external-link"><code>HarmonizR</code></a>
package, developed by <a href="https://doi.org/10.1038/s41467-022-31007-x" class="external-link">H.Voß &amp;
S.Schlumbohm (<em>Nat.Commun.</em>, 2022)</a>. This tool besides
handling multiple experiments, allows for the batch correction of data
derived by the combination of both DIA (Data-Independent Acquisition)
and DDA (Data-Dependent Acquisition).</p>
<blockquote>
<p><code>HarmonizR</code> is not a required dependency, therefore if
this function is used and <code>HarmonizR</code> is not already
installed, a warning will indicate the requred installation. The package
can be installed by: <code>BiocManager::install("HarmonizR")</code>, or
alternatively
<code>devtools::install_github("https://github.com/SimonSchlumbohm/HarmonizR/", subdir = "HarmonizR")</code>.</p>
</blockquote>
<p>To use the <code>harmonize.batches</code> is sufficient to provide a
<code>DEprot</code> object containing a combined table of multiple
experiments and indicate the identifier of a column in the metadata
table which corresponds to the batch annotation. The result is a DEprot
object with the same structure described in the previous paragraph. <br>
Here, for simplicity, we will add manually a column to the metadata
table with “dummy” batches.</p>
<p><br></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Adding batch column to the metadata table</span></span>
<span><span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">batch</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"batch_1"</span>,<span class="st">"batch_2"</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">6</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/get.metadata.html">get.metadata</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">)</span></span></code></pre></div>
<pre><code>&gt;              column.id           sample.id cell  condition    combined.id
&gt; 1         BCa_FBS_rep1        BCa_FBS_rep1  BCa        FBS        BCa_FBS
&gt; 2     BCa_6h.DMSO_rep1    BCa_6h.DMSO_rep1  BCa    6h.DMSO    BCa_6h.DMSO
&gt; 3  BCa_6h.10nM.E2_rep1 BCa_6h.10nM.E2_rep1  BCa 6h.10nM.E2 BCa_6h.10nM.E2
&gt; 4         BCa_FBS_rep2        BCa_FBS_rep2  BCa        FBS        BCa_FBS
&gt; 5     BCa_6h.DMSO_rep2    BCa_6h.DMSO_rep2  BCa    6h.DMSO    BCa_6h.DMSO
&gt; 6  BCa_6h.10nM.E2_rep2 BCa_6h.10nM.E2_rep2  BCa 6h.10nM.E2 BCa_6h.10nM.E2
&gt; 7         BCa_FBS_rep3        BCa_FBS_rep3  BCa        FBS        BCa_FBS
&gt; 8     BCa_6h.DMSO_rep3    BCa_6h.DMSO_rep3  BCa    6h.DMSO    BCa_6h.DMSO
&gt; 9  BCa_6h.10nM.E2_rep3 BCa_6h.10nM.E2_rep3  BCa 6h.10nM.E2 BCa_6h.10nM.E2
&gt; 10        BCa_FBS_rep4        BCa_FBS_rep4  BCa        FBS        BCa_FBS
&gt; 11    BCa_6h.DMSO_rep4    BCa_6h.DMSO_rep4  BCa    6h.DMSO    BCa_6h.DMSO
&gt; 12 BCa_6h.10nM.E2_rep4 BCa_6h.10nM.E2_rep4  BCa 6h.10nM.E2 BCa_6h.10nM.E2
&gt;    replicate old.column.id   batch
&gt; 1       rep1      Sample_A batch_1
&gt; 2       rep1      Sample_B batch_1
&gt; 3       rep1      Sample_C batch_1
&gt; 4       rep2      Sample_D batch_1
&gt; 5       rep2      Sample_E batch_1
&gt; 6       rep2      Sample_F batch_1
&gt; 7       rep3      Sample_G batch_2
&gt; 8       rep3      Sample_H batch_2
&gt; 9       rep3      Sample_I batch_2
&gt; 10      rep4      Sample_J batch_2
&gt; 11      rep4      Sample_K batch_2
&gt; 12      rep4      Sample_L batch_2</code></pre>
<p><br></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## batch correction</span></span>
<span><span class="va">dpo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/harmonize.batches.html">harmonize.batches</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                         batch.column <span class="op">=</span> <span class="st">"batch"</span>,</span>
<span>                         cores <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="imputation">
<strong>Data imputation</strong><a class="anchor" aria-label="anchor" href="#imputation"></a>
</h2>
<p>Often many <code>NA</code>/<code>NaN</code> values are present in the
LFQ tables due to the technical limitations of the protein detection by
Mass Spectrometry (MS) experiments.</p>
<p>Here we provide the possibility to use 4 different methods for the
imputation the data:</p>
<ul>
<li>
<a href="https://www.rdocumentation.org/packages/missForest/" class="external-link"><code>missForest</code></a>:
developed by <a href="https://doi.org/10.1093/bioinformatics/btr597" class="external-link">DJ.Stekhoven &amp;
P.Buehlmann (<em>Bioinformatics</em>, 2012)</a>, this tool will impute
the NaN and assign and estimated value. It also yields an out-of-bag
(OOB) imputation error estimate (general, or per each sample). Moreover,
it can be run parallel to save computation time (both examples reported
here after).</li>
<li>k-Nearest Neighbors (kNN) algorithm</li>
<li>Local Least Squares (LLS) imputation, using the pearson correlation
coefficient</li>
<li>Singular Value Decomposition (SVD)</li>
</ul>
<p><br></p>
<p>Of note, there are cases in which the missing data are not due to the
random but rather consequent to real biological effects (i.e.,
knock-down, knock-out, degron systems and PROTACs, etc.). This may lead
to the issue that a protein completely missing in all the replicates of
an individual condition will be imputed with relatively high numbers.
This would ultimately lead to the incorrect definition of the
differential status of this protein. To overcome this possibility, it is
possible to use the function <code>randomize.missing.values</code> to
assign random values from the bottom distribution of the full dataset.
The user can define a percentage threshold of minimal missing values
within the group; for instance if the threshold is 75% and in each group
there are 4 samples, it means that only proteins with at least 3 out of
4 missing values will be imputed. It will be sufficient to indicate the
ID of a column belonging to the metadata which indicates the
groups/conditions and the percentage of distribution from which select
random numbers (“tail.percentage”).</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dpo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/randomize.missing.values.html">randomize.missing.values</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                                group.column <span class="op">=</span> <span class="st">"combined.id"</span>,</span>
<span>                                percentage.missing <span class="op">=</span> <span class="fl">100</span>, <span class="co"># completely missing</span></span>
<span>                                tail.percentage <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span></code></pre></div>
<blockquote>
<p>Using the <code>"column.id"</code> as <em>group.column</em> is
equivalent to impute all the data with this method.</p>
</blockquote>
<p><br></p>
<p>Hereafter, an example in how to use the <code>missForest</code>
method to impute the data:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Without parallelization</span></span>
<span><span class="va">dpo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute.counts.html">impute.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                     method <span class="op">=</span> <span class="st">"missForest"</span>,</span>
<span>                     use.normalized.data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     missForest.max.iterations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                     missForest.variable.wise.OOBerror <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## With parallelization</span></span>
<span><span class="va">dpo</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/impute.counts.html">impute.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                     method <span class="op">=</span> <span class="st">"missForest"</span>,</span>
<span>                     use.normalized.data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     missForest.max.iterations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>                     missForest.variable.wise.OOBerror <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     missForest.cores <span class="op">=</span> <span class="fl">10</span>,</span>
<span>                     missForest.parallel.mode <span class="op">=</span> <span class="st">"variables"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dpo</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">@</span><span class="va">imputation</span><span class="op">$</span><span class="va">OOBerror</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">@</span><span class="va">imputation</span><span class="op">[</span><span class="op">-</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code>&gt; DEprot object:
&gt;            Samples:  12
&gt;           Proteins:  13239
&gt;   Counts available:  raw, normalized, imputed
&gt; Log transformation:  log2
&gt;   Metadata columns:  column.id, sample.id, cell, condition, combined.id, replicate, old.column.id</code></pre>
<pre><code>&gt;        BCa_FBS_rep1    BCa_6h.DMSO_rep1 BCa_6h.10nM.E2_rep1        BCa_FBS_rep2 
&gt;                   0                   0                   0                   0 
&gt;    BCa_6h.DMSO_rep2 BCa_6h.10nM.E2_rep2 
&gt;                   0                   0</code></pre>
<table class="table">
<colgroup>
<col width="15%">
<col width="21%">
<col width="30%">
<col width="8%">
<col width="23%">
</colgroup>
<thead><tr class="header">
<th align="left">method</th>
<th align="right">max.iterations</th>
<th align="left">parallelization.mode</th>
<th align="right">cores</th>
<th align="left">processing.time</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">missForest</td>
<td align="right">100</td>
<td align="left">variables</td>
<td align="right">12</td>
<td align="left">3.02 mins</td>
</tr></tbody>
</table>
<p><br></p>
<p>Also in this case a box/violin plot with the corresponding imputed
LFQ values per each sample is generated and stored in a vector slot.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.raw</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.norm</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">boxplot.imputed</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/plot_imputed-1.png" width="864" style="display: block; margin: auto;"></p>
<p><br></p>
<hr>
</div>
<div class="section level2">
<h2 id="pcacorrelation">
<strong>Sample similarities</strong><a class="anchor" aria-label="anchor" href="#pcacorrelation"></a>
</h2>
<div class="section level3">
<h3 id="pca">Principal Component Analyses (PCA)<a class="anchor" aria-label="anchor" href="#pca"></a>
</h3>
<p>PCA can be performed in order to perform a dimensional reduction and
determine which factor explains the variability of the samples.<br><code>DEprot</code> includes function dedicated to this aim and
specifically build to work with <code>DEprot</code> objects.</p>
<p>Notice that, even if the data are not log-transformed,
<code>perform.PCA</code> will do it before performing the analyses.</p>
<div class="section level4">
<h4 id="compute-pcs">Compute PCs<a class="anchor" aria-label="anchor" href="#compute-pcs"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform the analyses (DEprot.PCA object)</span></span>
<span><span class="va">PCA</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/perform.PCA.html">perform.PCA</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                   which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span> <span class="co"># possible: raw, normalized, imputed</span></span></code></pre></div>
<p>The <code>DEprot.PCA</code> object contains the following slots:</p>
<table class="table">
<colgroup>
<col width="16%">
<col width="83%">
</colgroup>
<thead><tr class="header">
<th align="right">Slot</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><em>PCA.metadata</em></td>
<td align="left">metadata of the samples used in the PCA (subset of the
original <code>DEprot@metadata</code>)</td>
</tr>
<tr class="even">
<td align="right"><em>sample.subset</em></td>
<td align="left">vector containing the list of samples analyzed</td>
</tr>
<tr class="odd">
<td align="right"><em>data.used</em></td>
<td align="left">vector indicating the type of counts used (imputed,
normalized, raw)</td>
</tr>
<tr class="even">
<td align="right"><em>prcomp</em></td>
<td align="left">object of class <code>prcomp</code> corresponding to
the full PCA output</td>
</tr>
<tr class="odd">
<td align="right"><em>PCs</em></td>
<td align="left">
<code>data.frame</code> combining the PC scores and the
metadata table, useful for replotting</td>
</tr>
<tr class="even">
<td align="right"><em>importance</em></td>
<td align="left">statistical summary table for the PCA analyses per each
PC</td>
</tr>
<tr class="odd">
<td align="right"><em>cumulative.PC.plot</em></td>
<td align="left">
<code>ggplot</code> object corresponding to out put of
<code>plot.PC.cumulative</code> for this object</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div class="section level4">
<h4 id="visualize-pcas">Visualize PCAs<a class="anchor" aria-label="anchor" href="#visualize-pcas"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot cumulative variance of all PCs</span></span>
<span><span class="co">#### equivalent to `PCA@cumulative.PC.plot`</span></span>
<span><span class="fu"><a href="../reference/plot.PC.cumulative.html">plot.PC.cumulative</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA</span>,</span>
<span>                   bar.color <span class="op">=</span> <span class="st">"steelblue"</span>,</span>
<span>                   line.color <span class="op">=</span> <span class="st">"navyblue"</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/run_cumulative-1.png" width="700" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot PC scatters</span></span>
<span><span class="va">PC_1.2</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.PC.scatter.html">plot.PC.scatter</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA</span>,</span>
<span>                  PC.x <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  PC.y <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                  color.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                  shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                  label.column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  plot.zero.lines <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="va">PC_2.3</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.PC.scatter.html">plot.PC.scatter</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA</span>,</span>
<span>                  PC.x <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                  PC.y <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                  color.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                  shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                  label.column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  plot.zero.lines <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">PC_1.2</span>, <span class="va">PC_2.3</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/run_PCA-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br></p>
<p>Alternatively, the combination of PC1.2 and PC2.3 can be
automatically generated using the function
<code>plot.PC.scatter.123</code>:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/plot.PC.scatter.123.html">plot.PC.scatter.123</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA</span>,</span>
<span>                    color.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                    shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                    label.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                    dot.colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6h.10nM.E2"</span> <span class="op">=</span> <span class="st">"indianred"</span>,</span>
<span>                                   <span class="st">"6h.DMSO"</span> <span class="op">=</span> <span class="st">"steelblue"</span>,</span>
<span>                                   <span class="st">"FBS"</span> <span class="op">=</span>  <span class="st">"forestgreen"</span><span class="op">)</span>,</span>
<span>                    plot.zero.line.y.12 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    plot.zero.line.x.12 <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                    plot.zero.line.y.23 <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    plot.zero.line.x.23 <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/unnamed-chunk-4-1.png" width="700" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="analyze-pcs-on-a-sample-subset">Analyze PCs on a sample subset<a class="anchor" aria-label="anchor" href="#analyze-pcs-on-a-sample-subset"></a>
</h4>
<p>These analyses can also be performed for a subset of samples by
indicated the sample names of interest.<br> In the example below we
will use only the sample in which the estrogen receptor is active (E2
and FBS conditions).</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Perform the analyses (DEprot.PCA object)</span></span>
<span><span class="va">PCA.fbs.e2</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/perform.PCA.html">perform.PCA</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>              sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"E2|FBS"</span>,</span>
<span>                                                           <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>              which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Plot cumulative variance of all PCs</span></span>
<span><span class="fu"><a href="../reference/plot.PC.cumulative.html">plot.PC.cumulative</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA.fbs.e2</span>,</span>
<span>                   bar.color <span class="op">=</span> <span class="st">"indianred"</span>,</span>
<span>                   line.color <span class="op">=</span> <span class="st">"firebrick4"</span>,</span>
<span>                   title <span class="op">=</span> <span class="st">"**Only ERa active**"</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/make_PCA_subset-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plot PC scatters</span></span>
<span><span class="va">PC.fbs.e2_1.2</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.PC.scatter.html">plot.PC.scatter</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA.fbs.e2</span>,</span>
<span>                  PC.x <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                  PC.y <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                  color.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                  shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                  label.column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  plot.zero.lines <span class="op">=</span> <span class="cn">F</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">geom_hline</span><span class="op">(</span>yintercept <span class="op">=</span> <span class="fl">0</span>, color <span class="op">=</span> <span class="st">"gray"</span>, linetype <span class="op">=</span> <span class="st">"dashed"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">theme</span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span>  </span>
<span>  </span>
<span><span class="va">PC.fbs.e2_2.3</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.PC.scatter.html">plot.PC.scatter</a></span><span class="op">(</span>DEprot.PCA.object <span class="op">=</span> <span class="va">PCA.fbs.e2</span>,</span>
<span>                  PC.x <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                  PC.y <span class="op">=</span> <span class="fl">3</span>,</span>
<span>                  color.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                  shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                  label.column <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>                  plot.zero.lines <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">PC.fbs.e2_1.2</span>, <span class="va">PC.fbs.e2_2.3</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/make_PCA_scatters_subset-1.png" width="768" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
</div>
<div class="section level3">
<h3 id="correlataion">Correlations<a class="anchor" aria-label="anchor" href="#correlataion"></a>
</h3>
<p>Another method to define the sample clustering/groups, is the overall
correlation between the samples.<br> Hierarchical clustering is
performed using the <code>1 - correlation</code> values, since the
hierarchical clustering algorithm is based on dissimilarities while the
correlations are an index of similarity.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">corr.all.samples</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.correlation.heatmap.html">plot.correlation.heatmap</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                           which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                           palette <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">mako</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, begin <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span>,</span>
<span>                           correlation.scale.limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                           correlation.method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>                           plot.subtitle <span class="op">=</span> <span class="st">"All samples"</span>,</span>
<span>                           display.values <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">corr.all.samples</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/make_correlation_all-1.png" width="864" style="display: block; margin: auto;"></p>
<p>Also in this case the sample correlation can be computed for a subset
of samples as shown before for the <a href="#pca">PCAs</a>.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">corr.ERa.active</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/plot.correlation.heatmap.html">plot.correlation.heatmap</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                           which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                           sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grepl</a></span><span class="op">(</span><span class="st">"E2|FBS"</span>,</span>
<span>                                                                        <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                           palette <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">magma</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span>, begin <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span>,</span>
<span>                           correlation.scale.limits <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.9</span>,<span class="fl">1</span><span class="op">)</span>,</span>
<span>                           correlation.method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>                           plot.subtitle <span class="op">=</span> <span class="st">"Only ERa active"</span>,</span>
<span>                           clustering.method <span class="op">=</span> <span class="st">"complete"</span>,</span>
<span>                           display.values <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">corr.ERa.active</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/make_correlation_subset-1.png" width="700" style="display: block; margin: auto;"></p>
<p>The <code>DEprot.correlation</code> correlation object contains the
following slots:</p>
<table class="table">
<colgroup>
<col width="14%">
<col width="85%">
</colgroup>
<thead><tr class="header">
<th align="right">Slot</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><em>heatmap</em></td>
<td align="left">
<code>ggplot</code> object corresponding to the
correlation heatmap</td>
</tr>
<tr class="even">
<td align="right"><em>corr.metadata</em></td>
<td align="left">metadata of the samples used in the correlation (subset
of the original <code>DEprot@metadata</code>)</td>
</tr>
<tr class="odd">
<td align="right"><em>sample.subset</em></td>
<td align="left">vector containing the list of samples analyzed</td>
</tr>
<tr class="even">
<td align="right"><em>data.used</em></td>
<td align="left">vector indicating the type of counts used (imputed,
normalized, raw)</td>
</tr>
<tr class="odd">
<td align="right"><em>corr.matrix</em></td>
<td align="left">the correlation matrix on which the heatmap is base
on</td>
</tr>
<tr class="even">
<td align="right"><em>distance</em></td>
<td align="left">object of class <code>dist</code> corresponding to the
output of <code>as.dist(1 - correlation.matrix)</code>
</td>
</tr>
<tr class="odd">
<td align="right"><em>cluster</em></td>
<td align="left">
<code>hclust</code> object generated by
<code>hclust(d = as.dist(1 - correlation.matrix), method = clustering.method)</code>
</td>
</tr>
</tbody>
</table>
<p><br></p>
<hr>
</div>
</div>
<div class="section level2">
<h2 id="diffexp">
<strong>Differential Expression (DE) analyses</strong><a class="anchor" aria-label="anchor" href="#diffexp"></a>
</h2>
<p>Differential expression analyses between two conditions can be
performed employing two different methods using the functions
<code>diff.analyses</code> and <code>diff.analyses.limma</code>. With
the first (<code>diff.analyses</code>) the conditions will be compared
two-by-two (individual t-/Wilcox tests), while with the second
(<code>diff.analyses.limma</code>) the <code>limma</code> package is
used to fit the data on a linear model.</p>
<p>In both cases, it is sufficient to provide a list of 3-elements
vectors. The latter, should indicate any column of the metadata table
(grouping factor) and two values (groups) to compare within this column.
The first variable will be numerator and the second the denominator of
the fold change:
<code>c("group.column", "condition.A", "condition.B")</code>, FoldChange
= group A/B.</p>
<p>When a replicate column is available, it is possible to run the
analyses in <em>paired</em> mode. When using <code>diff.analyses</code>,
it is sufficient to indicate the column from the metadata table that
corresponds to the replicate identifiers (in our example
<code>replicate.column = "replicate"</code>) and set
<code>paired.test = TRUE</code>. For each contrast it will be verified
that replicate identifiers are not repeated within a group and, that
replicate number and identifiers coincide between the two groups. By the
default analyses ignore the replicates. <br> If
<code>diff.analyses.limma</code> is used instead, the correlation
between replicates will be estimated and incorporated in the fitting
model. In this case indicate the replicate column identifier (in our
example <code>replicate.column = "replicate"</code>) and set
<code>include.rep.model = TRUE</code>.</p>
<blockquote>
<p>In the following example we will compare <em>6h.10nM.E2</em> vs
<em>6h.DMSO</em>, and <em>6h.10nM.E2</em> vs <em>FBS</em>. Both paired
and unpaired examples are shown, but further analyses are based on the
paired results only (multi t-test mode).</p>
</blockquote>
<p><br></p>
<div class="section level3">
<h3 id="differential-analyses-algorithms">Differential analyses algorithms<a class="anchor" aria-label="anchor" href="#differential-analyses-algorithms"></a>
</h3>
<div class="section level4">
<h4 id="limma-mode">
<code>limma</code> mode<a class="anchor" aria-label="anchor" href="#limma-mode"></a>
</h4>
<div class="section level5">
<h5 id="verify-normality-of-the-data">Verify normality of the data<a class="anchor" aria-label="anchor" href="#verify-normality-of-the-data"></a>
</h5>
<p>The main assumption to run <code>limma</code> is that the data are
normally distributed. For this reason is good practice verify that the
data are normally distributed for each sample. This can be accomplished
using the function <code>check.normality</code>. It will return a S4
vector composed by:</p>
<ul>
<li>
<code>norm.statement</code>: logical value indicating whether all
the samples are normally distributed (TRUE) or not (FALSE);</li>
<li>
<code>norm.AD.tests</code>: list of Anderson-Darling normality test
outputs (statistics, p-value, etc;</li>
<li>
<code>qqplots</code>: a list of Q-Q plots (ggplot objects) showing
the “correlation” between the distribution of your data and the
theoretical normal distribution</li>
<li>
<code>densities</code>: a list of density curves (ggplot objects)
depicted the distribution of your data and the overlay with the
theoretical normal distribution.</li>
</ul>
<blockquote>
<p>When a sample is not normally distributed (p-value of the AD
normality test &lt; p.threshold) the plot will be red instead of
black/blue. Furthermore, a message will tell if all samples are normally
distributed; and if not, which ones are not normally distributed.</p>
</blockquote>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>normality <span class="ot">&lt;-</span> <span class="fu">check.normality</span>(<span class="at">DEprot.object =</span> dpo,</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>                             <span class="at">p.threshold =</span> <span class="fl">0.05</span>,</span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>                             <span class="at">which.data =</span> <span class="st">"imputed"</span>,</span>
<span id="cb28-4"><a href="#cb28-4" tabindex="-1"></a>                             <span class="at">verbose =</span> T)</span>
<span id="cb28-5"><a href="#cb28-5" tabindex="-1"></a><span class="sc">&gt;</span> All samples display a normal distribution.</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## example of Q-Q and density plots</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">normality</span><span class="op">@</span><span class="va">qqplots</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>, <span class="va">normality</span><span class="op">@</span><span class="va">densities</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/normality_test_plots-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level5">
<h5 id="run-analyses-with-limma">Run analyses with <code>limma</code><a class="anchor" aria-label="anchor" href="#run-analyses-with-limma"></a>
</h5>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Unpaired test</span></span>
<span><span class="va">dpo_analyses</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diff.analyses.limma.html">diff.analyses.limma</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                                    contrast.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"6h.DMSO"</span><span class="op">)</span>,</span>
<span>                                                         <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                    linear.FC.th <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                    padj.th <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                    padj.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                                    fitting.method <span class="op">=</span> <span class="st">"ls"</span>,</span>
<span>                                    which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Paired test</span></span>
<span><span class="va">dpo_analyses</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diff.analyses.limma.html">diff.analyses.limma</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                                    contrast.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"6h.DMSO"</span><span class="op">)</span>,</span>
<span>                                                         <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                                    replicate.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                                    include.rep.model <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                    linear.FC.th <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                    padj.th <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                                    padj.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                                    fitting.method <span class="op">=</span> <span class="st">"ls"</span>,</span>
<span>                                    which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level4">
<h4 id="multiple-t-test-mode">Multiple t-test mode<a class="anchor" aria-label="anchor" href="#multiple-t-test-mode"></a>
</h4>
<blockquote>
<p>Next paragraphs use these results, in paired mode</p>
</blockquote>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Unpaired test</span></span>
<span><span class="va">dpo_analyses</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diff.analyses.html">diff.analyses</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                              contrast.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"6h.DMSO"</span><span class="op">)</span>,</span>
<span>                                                   <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                              linear.FC.th <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                              padj.th <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                              padj.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                              stat.test <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>                              which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Paired test</span></span>
<span><span class="va">dpo_analyses</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/diff.analyses.html">diff.analyses</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                              contrast.list <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"6h.DMSO"</span><span class="op">)</span>,</span>
<span>                                                   <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"condition"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span><span class="op">)</span>,</span>
<span>                              replicate.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                              paired.test <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                              linear.FC.th <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                              padj.th <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                              padj.method <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                              stat.test <span class="op">=</span> <span class="st">"t.test"</span>,</span>
<span>                              which.data <span class="op">=</span> <span class="st">"imputed"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dpo_analyses</span></span></code></pre></div>
<pre><code>&gt; DEprot.analyses object:
&gt;           Counts used:  imputed
&gt; Fold Change threshold:  2 (linear)
&gt; FC unresponsive range:  [0.9090909,1.1] (linear)
&gt;        padj threshold:  0.05 (linear)
&gt;           padj method:  BH
&gt; 
&gt; 
&gt; Differential results summary:
&gt;                        contrast.id group.factor     group1  group2 paired.test
&gt; 1 condition: 6h.10nM.E2 vs 6h.DMSO    condition 6h.10nM.E2 6h.DMSO        TRUE
&gt; 2 condition: 6h.10nM.E2 vs 6h.DMSO    condition 6h.10nM.E2 6h.DMSO        TRUE
&gt; 3 condition: 6h.10nM.E2 vs 6h.DMSO    condition 6h.10nM.E2 6h.DMSO        TRUE
&gt; 4 condition: 6h.10nM.E2 vs 6h.DMSO    condition 6h.10nM.E2 6h.DMSO        TRUE
&gt; 5     condition: 6h.10nM.E2 vs FBS    condition 6h.10nM.E2     FBS        TRUE
&gt; 6     condition: 6h.10nM.E2 vs FBS    condition 6h.10nM.E2     FBS        TRUE
&gt; 7     condition: 6h.10nM.E2 vs FBS    condition 6h.10nM.E2     FBS        TRUE
&gt; 8     condition: 6h.10nM.E2 vs FBS    condition 6h.10nM.E2     FBS        TRUE
&gt;    diff.status     n median.FoldChange
&gt; 1      6h.DMSO     0                NA
&gt; 2   6h.10nM.E2     0                NA
&gt; 3 unresponsive  8557       0.004011163
&gt; 4         null  4682       0.150392374
&gt; 5          FBS   272      -1.273486884
&gt; 6   6h.10nM.E2   299       1.260394829
&gt; 7 unresponsive  2358      -0.010138355
&gt; 8         null 10310      -0.188567885</code></pre>
<p>The summary can be collected by using the generic function
<code>summary</code>:</p>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">diff.analyses_summary</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">dpo</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level3">
<h3 id="de-results">DE results<a class="anchor" aria-label="anchor" href="#de-results"></a>
</h3>
<p>The output will be a <code>DEprot.analyses</code> object. This class
is similar to the base <code>DEprot</code> one, however 3 slots are now
available:</p>
<ul>
<li>
<em>contrasts</em>: corresponds to the list used to define the
contrasts, but includes also the IDs of the counts matrix belonging to
each subgroup and whether the statistical test was performed in
<em>paired</em> mode.</li>
<li>
<em>differential.analyses.params</em>: a list containing the core
parameters used for the differential expression analyses.</li>
<li>
<em>analyses.result.list</em>: a list with an element for each
contrast including all the results of the differential analyses (see
below for details).</li>
</ul>
<p>The <code>analyses.result.list</code>, for each contrast, stores a
list with the following elements:</p>
<table class="table">
<colgroup>
<col width="10%">
<col width="89%">
</colgroup>
<thead><tr class="header">
<th align="right">Element</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><em>results</em></td>
<td align="left">a data.frame containing the results of the analyses;
includes average expression of each group, basemean, foldchange, pvalue
and p.adj, differential.status</td>
</tr>
<tr class="even">
<td align="right"><em>n.diff</em></td>
<td align="left">a summary table showing the number of proteins in each
differential expression status (up/down/unresponsive, null)</td>
</tr>
<tr class="odd">
<td align="right"><em>PCA.data</em></td>
<td align="left">output of <code>perform.PCA</code> for the subset of
samples analyzed in a specific contrast</td>
</tr>
<tr class="even">
<td align="right"><em>PCA.plots</em></td>
<td align="left">combination of 3 plots: scatter PC1-vs-PC2, scatter
PC2-vs-PC3, and cumulative bar plot</td>
</tr>
<tr class="odd">
<td align="right"><em>correlations</em></td>
<td align="left">combination of Pearson and Spearman correlation
heatmaps (obtained by <code>plot.correlation.heatmap</code>) for the
subset of samples analyzed in a specific contrast</td>
</tr>
<tr class="even">
<td align="right"><em>volcano</em></td>
<td align="left">volcano plot showing the log2(FoldChange) x
-log10(p.adjusted) of differential expression results; it can be
regenerated using <code>plot.volcano</code>
</td>
</tr>
<tr class="odd">
<td align="right"><em>MA.plot</em></td>
<td align="left">MA-plot showing the log2(basemean) x log2(FoldChange)
of differential expression results; it can be regenerated using
<code>plot.MA</code>
</td>
</tr>
</tbody>
</table>
<p>If the function <code>diff.analyses.limma</code> was used, an
additional slot called <code>limma.fit</code> will be available and will
contain the results of <code><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">limma::contrasts.fit</a></code>.</p>
<p><br></p>
<p>The full object can be automatically exported - including tables,
PCA, correlations, volcano and MA-plots, etc. - using the built-in
function <code>export.analyses</code>:</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/export.analyses.html">export.analyses</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>, output.folder <span class="op">=</span> <span class="st">"./export"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<div class="section level4">
<h4 id="de-table">DE table<a class="anchor" aria-label="anchor" href="#de-table"></a>
</h4>
<p>The table with the results of the differential analyses can be
retrieved directly from the list in the <code>DEprot.analyses</code>
object
(<code>dpo_analyses@analyses.result.list$contrast.id$results</code>) or
using the <code>get.results</code> function.</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Direct access</span></span>
<span><span class="va">results</span> <span class="op">=</span> <span class="va">dpo_analyses</span><span class="op">@</span><span class="va">analyses.result.list</span><span class="op">$</span><span class="va">condition_6h.10nM.E2.vs.6h.DMSO</span><span class="op">$</span><span class="va">results</span></span>
<span></span>
<span><span class="co">## Function</span></span>
<span><span class="va">results</span> <span class="op">=</span> <span class="fu"><a href="../reference/get.results.html">get.results</a></span><span class="op">(</span><span class="va">dpo_analyses</span>, contrast <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="7%">
<col width="10%">
<col width="16%">
<col width="14%">
<col width="25%">
<col width="7%">
<col width="7%">
<col width="10%">
</colgroup>
<thead><tr class="header">
<th align="left">prot.id</th>
<th align="right">basemean.log2</th>
<th align="right">log2.mean.6h.10nM.E2</th>
<th align="right">log2.mean.6h.DMSO</th>
<th align="right">log2.Fold_6h.10nM.E2.vs.6h.DMSO</th>
<th align="right">p.value</th>
<th align="right">padj</th>
<th align="left">diff.status</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">protein.1</td>
<td align="right">18.85442</td>
<td align="right">18.91504</td>
<td align="right">18.79380</td>
<td align="right">0.1212396</td>
<td align="right">0.6342736</td>
<td align="right">0.9189262</td>
<td align="left">unresponsive</td>
</tr>
<tr class="even">
<td align="left">protein.2</td>
<td align="right">23.07816</td>
<td align="right">23.08739</td>
<td align="right">23.06894</td>
<td align="right">0.0184512</td>
<td align="right">0.8673988</td>
<td align="right">0.9781933</td>
<td align="left">unresponsive</td>
</tr>
<tr class="odd">
<td align="left">protein.3</td>
<td align="right">15.68530</td>
<td align="right">15.65794</td>
<td align="right">15.71266</td>
<td align="right">-0.0547179</td>
<td align="right">0.7188334</td>
<td align="right">0.9405324</td>
<td align="left">unresponsive</td>
</tr>
<tr class="even">
<td align="left">protein.4</td>
<td align="right">21.14521</td>
<td align="right">21.09930</td>
<td align="right">21.19112</td>
<td align="right">-0.0918296</td>
<td align="right">0.2081734</td>
<td align="right">0.7869470</td>
<td align="left">unresponsive</td>
</tr>
<tr class="odd">
<td align="left">protein.5</td>
<td align="right">19.56671</td>
<td align="right">19.62718</td>
<td align="right">19.50624</td>
<td align="right">0.1209374</td>
<td align="right">0.1169272</td>
<td align="right">0.7417323</td>
<td align="left">unresponsive</td>
</tr>
<tr class="even">
<td align="left">protein.6</td>
<td align="right">15.31226</td>
<td align="right">15.32551</td>
<td align="right">15.29901</td>
<td align="right">0.0264997</td>
<td align="right">0.6433742</td>
<td align="right">0.9227061</td>
<td align="left">unresponsive</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div class="section level4">
<h4 id="pca-and-correlation-within-the-comparison">PCA and correlation within the comparison<a class="anchor" aria-label="anchor" href="#pca-and-correlation-within-the-comparison"></a>
</h4>
<p>The <code>DEprot.analyses</code> object includes PCA and correlation
analyses of the samples involved in the contrast.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dpo_analyses</span><span class="op">@</span><span class="va">analyses.result.list</span><span class="op">$</span><span class="va">condition_6h.10nM.E2.vs.6h.DMSO</span><span class="op">$</span><span class="va">PCA.plots</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/DE_PCA_scatters_replotting-1.png" width="768" style="display: block; margin: auto;"></p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">dpo_analyses</span><span class="op">@</span><span class="va">analyses.result.list</span><span class="op">$</span><span class="va">condition_6h.10nM.E2.vs.6h.DMSO</span><span class="op">$</span><span class="va">correlations</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/DE_correlations-1.png" width="960" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level4">
<h4 id="visualize-de-analyses">Visualize DE analyses<a class="anchor" aria-label="anchor" href="#visualize-de-analyses"></a>
</h4>
<div class="section level5">
<h5 id="p-values-distributions">P-values distributions<a class="anchor" aria-label="anchor" href="#p-values-distributions"></a>
</h5>
<p>Good practice is to visualize the distribution of the p-values and
compare it with the adjusted ones in order to estimate the quality of
the statistical tools used. In the histograms, the horizontal dashed
line indicates the median of the frequency, while the vertical dotted
line demarcates the P-value threshold used for the differential
analyses.<br> This can be performed using the
<code>check.pvalues</code> function:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pval.distribution</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/check.pvalues.html">check.pvalues</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                                   contrast <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                                   histogram.binwidth <span class="op">=</span> <span class="fl">0.025</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pval.distribution</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/check_pvalues-1.png" width="768" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level5">
<h5 id="ma-and-volcano">MA and volcano<a class="anchor" aria-label="anchor" href="#ma-and-volcano"></a>
</h5>
<p>Differential expressed proteins can be visualized as either a volcano
plot or an MA-plot.<br> Both these plots are available in the
<code>dpo_analyses@analyses.result.list$contrast.id</code> list, but can
also be generated using the functions <code>plot.volcano</code> and
<code>plot.MA</code>.</p>
<blockquote>
<p>Of note, if <code>use.uncorrected.pvalue = TRUE</code>, the normal
p-value will be used instead of the p.adjusted. In this case the
FoldChange and p-value thresholds are collected from the
<code>DEprot.analyses</code> object and reapplied to compute the new
differential status of the proteins.</p>
</blockquote>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">volcano</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot.volcano.html">plot.volcano</a></span><span class="op">(</span><span class="va">dpo_analyses</span>, contrast <span class="op">=</span> <span class="fl">1</span>, use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">MAplot</span> <span class="op">=</span> <span class="fu"><a href="../reference/plot.MA.html">plot.MA</a></span><span class="op">(</span><span class="va">dpo_analyses</span>, contrast <span class="op">=</span> <span class="fl">1</span>, use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">volcano</span>, <span class="va">MAplot</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/DE_volcano_MA-1.png" width="864" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level5">
<h5 id="correlation-between-contrasts">Correlation between contrasts<a class="anchor" aria-label="anchor" href="#correlation-between-contrasts"></a>
</h5>
<p>Using the function <code>contrast.scatter</code> it is possible to
visualize the log<sub>2</sub>(Fold Change) correlation between two
contrasts of a differential expression analyses object.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">contrast.scatter</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/contrast.scatter.html">contrast.scatter</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                   contrast.x <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                   contrast.y <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   regression.line.color <span class="op">=</span> <span class="st">"firebrick"</span>,</span>
<span>                   correlation.method <span class="op">=</span> <span class="st">"pearson"</span>,</span>
<span>                   add.foldchange.threshold <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                   symmetric.axes <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">contrast.scatter</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/contrast_scatter-1.png" width="480" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level5">
<h5 id="heatmaps">Heatmaps<a class="anchor" aria-label="anchor" href="#heatmaps"></a>
</h5>
<p>To plot results heatmaps there are two functions available:</p>
<ul>
<li>
<code>heatmap.counts</code>: allows for the plotting of raw,
normalized or imputed counts.</li>
<li>
<code>heatmap.contrasts</code>: allows for the plotting of
log<sub>2</sub>(FoldChange) values.</li>
</ul>
<p><br></p>
<div class="section level6">
<h6 id="counts">Counts<a class="anchor" aria-label="anchor" href="#counts"></a>
</h6>
<p>With the <code>heatmap.counts</code> function it is possible to
generate an heatmap for the raw/normalized/imputed counts. It is
possible to select only specific proteins and/or subset samples (from
<code>column.id</code> column in metadata table). If the object provided
is a <code>DEprot.analyses</code> object, it is also possible to
indicate to plot only differential proteins (or top.n differential
proteins) from a specific contrast. <br> The protein ranking
definition is based on the differential score, computed as:
<code>abs(log2(fold change)) * -log10(Padj)</code>.</p>
<p>The resulting object is of class <code>DEprot.counts.heatmap</code>,
and contains the ggplot-heatmap, the row cluster and the column
cluster.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Plotting from a DEprot object</span></span>
<span><span class="va">imputed_counts_heatmap</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/heatmap.counts.html">heatmap.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                 which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                 sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"6h"</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                 show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 protein.subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"protein.2295"</span>, <span class="st">"protein.304"</span>, <span class="st">"protein.657"</span>,</span>
<span>                                    <span class="st">"protein.2819"</span>, <span class="st">"protein.2168"</span>, <span class="st">"protein.10594"</span><span class="op">)</span>,</span>
<span>                 title <span class="op">=</span> <span class="st">"Imputed counts | protein and sample selection"</span><span class="op">)</span> </span>
<span></span>
<span></span>
<span></span>
<span><span class="co">## Plotting from a DEprot.analyses object</span></span>
<span><span class="co">## top 15 differential proteins from contrast 1</span></span>
<span><span class="va">imputed_counts_heatmap_diffProteins</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/heatmap.counts.html">heatmap.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                 which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                 contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 top.n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>                 palette <span class="op">=</span> <span class="fu">viridis</span><span class="fu">::</span><span class="fu"><a href="https://sjmgarnier.github.io/viridisLite/reference/viridis.html" class="external-link">mako</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fl">100</span>, direction <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span>,</span>
<span>                 cell.border.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"6h"</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                 use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 protein.names.pattern <span class="op">=</span> <span class="st">"protein[.]"</span>,</span>
<span>                 title <span class="op">=</span> <span class="st">"condition: **6h.10nM.E2** *vs* **6h.DMSO** (top 15) | Imputed counts"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Combine heatmaps</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">imputed_counts_heatmap</span><span class="op">@</span><span class="va">heatmap</span>,</span>
<span>                      <span class="va">imputed_counts_heatmap_diffProteins</span><span class="op">@</span><span class="va">heatmap</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/heatmap_counts-1.png" width="1248" style="display: block; margin: auto;"></p>
<p><br></p>
<p>Instead of “pure” counts, it is possible to compute Z-scores of the
values and plot a divergent heatmap.<br> In this case instead of a
palette, it will be required to indicate the color for the positive,
negative and 0 values.</p>
<p>We will apply this to the
<code>imputed_counts_heatmap_diffProteins</code> heatmap from the
previous example.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Z-score by row</span></span>
<span><span class="va">imputed_counts_heatmap_diffProteins_rowScaled</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/heatmap.counts.html">heatmap.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                 which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                 contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 top.n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>                 high.color <span class="op">=</span> <span class="st">"purple4"</span>,</span>
<span>                 low.color <span class="op">=</span> <span class="st">"darkorange"</span>,</span>
<span>                 mid.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 cell.border.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"6h"</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                 use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 scale <span class="op">=</span> <span class="st">"rows"</span>,</span>
<span>                 title <span class="op">=</span> <span class="st">"condition: **6h.10nM.E2** *vs* **6h.DMSO** (top 15)&lt;br&gt;Imputed counts Z-score"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Z-score by column</span></span>
<span><span class="va">imputed_counts_heatmap_diffProteins_columnScaled</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/heatmap.counts.html">heatmap.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                 which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                 contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 top.n <span class="op">=</span> <span class="fl">15</span>,</span>
<span>                 high.color <span class="op">=</span> <span class="st">"firebrick"</span>,</span>
<span>                 low.color <span class="op">=</span> <span class="st">"steelblue"</span>,</span>
<span>                 mid.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 cell.border.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 sample.subset <span class="op">=</span> <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"6h"</span>, <span class="va">dpo</span><span class="op">@</span><span class="va">metadata</span><span class="op">$</span><span class="va">column.id</span><span class="op">)</span><span class="op">]</span>,</span>
<span>                 use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 scale <span class="op">=</span> <span class="st">"columns"</span>,</span>
<span>                 title <span class="op">=</span> <span class="st">"condition: **6h.10nM.E2** *vs* **6h.DMSO** (top 15)&lt;br&gt;Imputed counts Z-score"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">## Combine heatmaps</span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">imputed_counts_heatmap_diffProteins_rowScaled</span><span class="op">@</span><span class="va">heatmap</span>,</span>
<span>                      <span class="va">imputed_counts_heatmap_diffProteins_columnScaled</span><span class="op">@</span><span class="va">heatmap</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/heatmap_counts_zscores-1.png" width="960" style="display: block; margin: auto;"></p>
<p><br></p>
<p>Ultimately, With this function is also possible to group data based
on the values of a metadata’s column. In this way it is possible to
average the values by the values of the defined column (e.g., the
replicates of a condition).</p>
<p>In the next example we will average the replicates of each condition
using the <code>combined.id</code> column form the metadata of the
<code>dpo_analyses</code> object.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">imputed_counts_heatmap_diffProteins_rowScaled_grouped.by.condition</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/heatmap.counts.html">heatmap.counts</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                 group.by.metadata.column <span class="op">=</span> <span class="st">"combined.id"</span>,</span>
<span>                 which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                 contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                 high.color <span class="op">=</span> <span class="st">"firebrick"</span>,</span>
<span>                 low.color <span class="op">=</span> <span class="st">"steelblue"</span>,</span>
<span>                 mid.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 cell.border.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                 show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 scale <span class="op">=</span> <span class="st">"rows"</span>,</span>
<span>                 title <span class="op">=</span> <span class="st">"condition: **6h.10nM.E2** *vs* **6h.DMSO** (all)&lt;br&gt;Imputed counts Z-score"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">imputed_counts_heatmap_diffProteins_rowScaled_grouped.by.condition</span><span class="op">@</span><span class="va">heatmap</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/heatmap_counts_groups-1.png" width="384" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level6">
<h6 id="fold-changes">Fold changes<a class="anchor" aria-label="anchor" href="#fold-changes"></a>
</h6>
<p>Instead of counts, it is possible to plot the fold changes of one or
multiple contrasts using the function <code>heatmap.contrasts</code>.
Only differential proteins are shown.<br> Also in this case it is
possible to show only the top <em>N</em> differential proteins. However
the top <em>N</em> of each contrast will be shown. This means than
<em>N</em> proteins might be shown.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">FC_heatmap</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/heatmap.contrasts.html">heatmap.contrasts</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                    contrasts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">)</span>,</span>
<span>                    top.n <span class="op">=</span> <span class="fl">20</span>,</span>
<span>                    high.color <span class="op">=</span> <span class="st">"#35978F"</span>,</span>
<span>                    low.color <span class="op">=</span> <span class="st">"#BF812D"</span>,</span>
<span>                    mid.color <span class="op">=</span> <span class="st">"white"</span>,</span>
<span>                    show.protein.names <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    protein.names.pattern <span class="op">=</span> <span class="st">"protein[.]"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">FC_heatmap</span><span class="op">@</span><span class="va">heatmap</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/heatmap_foldchanges-1.png" width="384" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
</div>
<div class="section level5">
<h5 id="upset-plot">Upset-plot<a class="anchor" aria-label="anchor" href="#upset-plot"></a>
</h5>
<p>To identify which differential proteins are common among multiple
comparisons, it is possible to generate an upset plot using the function
<code>plot.upset</code>. With this function all the intersections
between all the contrasts including in the provided object can be
visualized by co-occurrence. It is also possible to subset only specific
contrasts indicating a numeric vector in the flag
<code>sort.intersections</code>.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">upset.plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot.upset.html">plot.upset</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                         contrast.subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">2</span><span class="op">)</span>,</span>
<span>                         title <span class="op">=</span> <span class="st">"**My upset plot**"</span>,</span>
<span>                         sort.intersections <span class="op">=</span> <span class="st">"cardinality"</span>,</span>
<span>                         sort.sets <span class="op">=</span> <span class="st">"descending"</span>,</span>
<span>                         intersection.bar.color <span class="op">=</span> <span class="st">"navy"</span>,</span>
<span>                         setsize.bar.color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>                         show.counts <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                         height.ratio <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>                         width.ratio <span class="op">=</span> <span class="fl">0.4</span>,</span>
<span>                         use.uncorrected.pvalue <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="va">upset.plot</span>  <span class="co"># or upset.plot@upset</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/upset_plot-1.png" width="700" style="display: block; margin: auto;"></p>
<p><br></p>
<p>Besides the plot, also a <code>TRUE</code>/<code>FALSE</code> table
is returned. The latter shows for each condition if a protein has been
found to be differential. Only proteins that are differential in at
least one condition are shown.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">upset.plot</span><span class="op">@</span><span class="va">obs.matrix</span></span></code></pre></div>
<table class="table">
<caption><strong>Upset observations matrix</strong></caption>
<colgroup>
<col width="4%">
<col width="25%">
<col width="24%">
<col width="20%">
<col width="23%">
</colgroup>
<thead><tr class="header">
<th align="left">prot.id</th>
<th align="left">condition: 6h.10nM.E2 <em>vs</em> 6h.DMSO |
<strong>6h.10nM.E2</strong>
</th>
<th align="left">condition: 6h.10nM.E2 <em>vs</em> 6h.DMSO |
<strong>6h.DMSO</strong>
</th>
<th align="left">condition: 6h.10nM.E2 <em>vs</em> FBS |
<strong>FBS</strong>
</th>
<th align="left">condition: 6h.10nM.E2 <em>vs</em> FBS |
<strong>6h.10nM.E2</strong>
</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">protein.17</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="even">
<td align="left">protein.25</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="odd">
<td align="left">protein.44</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
<tr class="even">
<td align="left">protein.48</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
<td align="left">FALSE</td>
</tr>
<tr class="odd">
<td align="left">protein.53</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">FALSE</td>
<td align="left">TRUE</td>
</tr>
</tbody>
</table>
<p><br></p>
<hr>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="other-tools">
<strong>Other tools</strong><a class="anchor" aria-label="anchor" href="#other-tools"></a>
</h2>
<div class="section level3">
<h3 id="count-detected-proteins">Count detected proteins<a class="anchor" aria-label="anchor" href="#count-detected-proteins"></a>
</h3>
<p>In order to know how many proteins have been detected in each sample,
it is possible to use the function <code>protein.summary</code>. A
stacked bar plot will be generated per each sample. The input object
must belong to either <code>DEprot</code> or
<code>DEprot.analyses</code> classes.</p>
<blockquote>
<p><strong>NOTE</strong>: this plot is meaningfult only on raw and/or
unimputed normalized data. Indeed, estimation of missing values is not
possibile on the imputed tables.</p>
</blockquote>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">protein.counts</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/protein.summary.html">protein.summary</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                  n.labels <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>                  show.frequency <span class="op">=</span> <span class="cn">F</span>,</span>
<span>                  colors <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gray"</span>, <span class="st">"steelblue4"</span><span class="op">)</span>,</span>
<span>                  title <span class="op">=</span> <span class="st">"**# Proteins identified in each sample**"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">protein.counts</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/protein_summary-1.png" width="960" style="display: block; margin: auto;"></p>
<p><br></p>
<p>Another possibility is to study the protein co-detection by sample
groups. A group can be defined using any column of the metadata table.
In the output stacked bar plot, each color will show how many proteins
in how many samples within a group were detected.</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">protein.counts.byCondition</span> <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="../reference/protein.summary.html">protein.summary</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                  group.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                  n.labels <span class="op">=</span> <span class="st">"percentage"</span>,</span>
<span>                  show.frequency <span class="op">=</span> <span class="cn">T</span>,</span>
<span>                  x.label.angle <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                  title <span class="op">=</span> <span class="st">"**# Proteins identified per _Condition_**"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">protein.counts.byCondition</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/protein_summary_by_condition-1.png" width="384" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="individual-protein-expression">Individual protein expression<a class="anchor" aria-label="anchor" href="#individual-protein-expression"></a>
</h3>
<p>Using the function <code>expression.boxplot</code> is possible to
generate boxplots of the expression of individual proteins.<br> The
expression can be grouped by a column of the metadata using the
<code>group.by.metadata.column</code> parameter. Furthermore it is
possible to scale the values using Z-scores (parameter
<code>scale.expression = TRUE</code>).</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### raw expression</span></span>
<span><span class="va">protein.1733_raw</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/expression.boxplot.html">expression.boxplot</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                     protein.id <span class="op">=</span> <span class="st">"protein.1733"</span>,</span>
<span>                     which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                     shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                     group.by.metadata.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                     group.levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6h.DMSO"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span>,</span>
<span>                     scale.expression <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                     x.label.angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">### scaled expression</span></span>
<span><span class="va">protein.1733_scaled</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/expression.boxplot.html">expression.boxplot</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo</span>,</span>
<span>                     protein.id <span class="op">=</span> <span class="st">"protein.1733"</span>,</span>
<span>                     which.data <span class="op">=</span> <span class="st">"imputed"</span>,</span>
<span>                     shape.column <span class="op">=</span> <span class="st">"replicate"</span>,</span>
<span>                     group.by.metadata.column <span class="op">=</span> <span class="st">"condition"</span>,</span>
<span>                     group.levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"6h.DMSO"</span>, <span class="st">"6h.10nM.E2"</span>, <span class="st">"FBS"</span><span class="op">)</span>,</span>
<span>                     scale.expression <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                     x.label.angle <span class="op">=</span> <span class="fl">90</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">protein.1733_raw</span>, <span class="va">protein.1733_scaled</span>, nrow <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/expression_boxplot-1.png" width="672" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="filter-proteins">Filter proteins<a class="anchor" aria-label="anchor" href="#filter-proteins"></a>
</h3>
<p>For certain analyses it might be necessary to filter-out/keep only
certain specific protein (e.g., keep only nuclear proteins).<br>
With the function <code>filter.proteins</code> is possible to indicate a
vector containing the protein IDs (it must correspond to the row.names
of the counts table) and chose whether to <code>"keep"</code> or
<code>"remove"</code> these proteins.</p>
<p>This function works for both <code>DEprot</code> and
<code>DEprot.analyses</code> classes. The output will look exactly as
the original, however counts tables (raw, normalized, imputed), count
boxplots, PCAs, correlations, differential tables, differential protein
counts, violin and MA-plot will be recomputed.</p>
<blockquote>
<p><strong>NOTE</strong>: we discourage the filtering before
normalization/bacth correction and imputation, as well as before
limma-base differential analyses. In all these methods are based on the
assumption that the proteins ‘available’ are ‘all’ the proteins
available, which is not the case.</p>
</blockquote>
<p><br></p>
<div class="section level4">
<h4 id="filtering-examples">Filtering examples<a class="anchor" aria-label="anchor" href="#filtering-examples"></a>
</h4>
<p>Hereafter are shown two examples for keeping nuclear proteins or
removing cytoplasmic proteins.</p>
<blockquote>
<p>Notice that this is only an example and, since the protein IDs are
randomized, it would not applicable for the example dataset availble
with <code>DEprot</code>. Furthermore <code>DEprot</code> does not
include the packages <code>AnnotationDbi</code> and
<code>org.Hs.eg.db</code>, which can be installed throug the
<code>Bioconductor</code> portal.</p>
</blockquote>
<p><br></p>
<p><strong>KEEP NUCLEAR PROTEINS</strong></p>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nucleus</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va">org.Hs.eg.db</span>,</span>
<span>                                 keytype <span class="op">=</span> <span class="st">"GOALL"</span>,</span>
<span>                                 keys <span class="op">=</span> <span class="st">"GO:0005634"</span>, <span class="co">#nucleus</span></span>
<span>                                 columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"UNIPROT"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dpo_analyses_nuclear</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.proteins.html">filter.proteins</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                                        proteins <span class="op">=</span> <span class="va">nucleus</span><span class="op">$</span><span class="va">SYMBOL</span>,</span>
<span>                                        mode <span class="op">=</span> <span class="st">"keep"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
<p><strong>REMOVE CYTOPLASMIC PROTEINS</strong></p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cytoplasm</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="fu">org.Hs.eg.db</span><span class="fu">::</span><span class="va">org.Hs.eg.db</span>,</span>
<span>                                   keytype <span class="op">=</span> <span class="st">"GOALL"</span>,</span>
<span>                                   keys <span class="op">=</span> <span class="st">"GO:0005737"</span>, <span class="co">#cytoplasm</span></span>
<span>                                   columns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"SYMBOL"</span>, <span class="st">"UNIPROT"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dpo_analyses_nuclear</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter.proteins.html">filter.proteins</a></span><span class="op">(</span>DEprot.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                                        proteins <span class="op">=</span> <span class="va">cytoplasm</span><span class="op">$</span><span class="va">SYMBOL</span>,</span>
<span>                                        mode <span class="op">=</span> <span class="st">"remove"</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
</div>
<div class="section level3">
<h3 id="re-apply-thresholds-to-deprot-analyses-object">Re-apply thresholds to <code>DEprot.analyses</code> object<a class="anchor" aria-label="anchor" href="#re-apply-thresholds-to-deprot-analyses-object"></a>
</h3>
<p>In certain cases, when several contrasts are defined, the
differential analysis computation can be time consuming. On the other
hand, one might want to change the thresholds used to define the
differential status of the proteins. To avoid the re-computation of the
differentials from the beginning, the function
<code>reapply.thresholds</code> can be used to re-compute the
differential status, re-plot volcanos and MA-plots, and the n.diff
summary table.</p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" tabindex="-1"></a>dpo_analyses_fc1<span class="fl">.5</span> <span class="ot">=</span></span>
<span id="cb52-2"><a href="#cb52-2" tabindex="-1"></a>  <span class="fu">reapply.thresholds</span>(dpo_analyses,</span>
<span id="cb52-3"><a href="#cb52-3" tabindex="-1"></a>                     <span class="at">linear.FC =</span> <span class="fl">1.5</span>,</span>
<span id="cb52-4"><a href="#cb52-4" tabindex="-1"></a>                     <span class="at">p.adjusted =</span> <span class="fl">0.05</span>,</span>
<span id="cb52-5"><a href="#cb52-5" tabindex="-1"></a>                     <span class="at">linear.FC.unresp.range =</span> <span class="fu">c</span>(<span class="dv">1</span><span class="sc">/</span><span class="fl">1.1</span>, <span class="fl">1.1</span>),</span>
<span id="cb52-6"><a href="#cb52-6" tabindex="-1"></a>                     <span class="at">up.color =</span> <span class="st">"indianred"</span>,</span>
<span id="cb52-7"><a href="#cb52-7" tabindex="-1"></a>                     <span class="at">down.color =</span> <span class="st">"steelblue"</span>,</span>
<span id="cb52-8"><a href="#cb52-8" tabindex="-1"></a>                     <span class="at">unresponsive.color =</span> <span class="st">"purple"</span>,</span>
<span id="cb52-9"><a href="#cb52-9" tabindex="-1"></a>                     <span class="at">null.color =</span> <span class="st">"gray"</span>)</span>
<span id="cb52-10"><a href="#cb52-10" tabindex="-1"></a></span>
<span id="cb52-11"><a href="#cb52-11" tabindex="-1"></a><span class="fu">summary</span>(dpo_analyses_fc1<span class="fl">.5</span>)</span>
<span id="cb52-12"><a href="#cb52-12" tabindex="-1"></a><span class="sc">&gt;</span>                        contrast.id group.factor     group1  group2 paired.test</span>
<span id="cb52-13"><a href="#cb52-13" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">1</span> condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs <span class="dv">6</span>h.DMSO    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 <span class="dv">6</span>h.DMSO        <span class="cn">TRUE</span></span>
<span id="cb52-14"><a href="#cb52-14" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">2</span> condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs <span class="dv">6</span>h.DMSO    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 <span class="dv">6</span>h.DMSO        <span class="cn">TRUE</span></span>
<span id="cb52-15"><a href="#cb52-15" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">3</span> condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs <span class="dv">6</span>h.DMSO    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 <span class="dv">6</span>h.DMSO        <span class="cn">TRUE</span></span>
<span id="cb52-16"><a href="#cb52-16" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">4</span> condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs <span class="dv">6</span>h.DMSO    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 <span class="dv">6</span>h.DMSO        <span class="cn">TRUE</span></span>
<span id="cb52-17"><a href="#cb52-17" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">5</span>     condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs FBS    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2     FBS        <span class="cn">TRUE</span></span>
<span id="cb52-18"><a href="#cb52-18" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">6</span>     condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs FBS    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2     FBS        <span class="cn">TRUE</span></span>
<span id="cb52-19"><a href="#cb52-19" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">7</span>     condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs FBS    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2     FBS        <span class="cn">TRUE</span></span>
<span id="cb52-20"><a href="#cb52-20" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">8</span>     condition<span class="sc">:</span> <span class="dv">6</span>h<span class="fl">.10</span>nM.E2 vs FBS    condition <span class="dv">6</span>h<span class="fl">.10</span>nM.E2     FBS        <span class="cn">TRUE</span></span>
<span id="cb52-21"><a href="#cb52-21" tabindex="-1"></a><span class="sc">&gt;</span>    diff.status    n median.FoldChange</span>
<span id="cb52-22"><a href="#cb52-22" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">1</span>      <span class="dv">6</span>h.DMSO    <span class="dv">0</span>                <span class="cn">NA</span></span>
<span id="cb52-23"><a href="#cb52-23" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">2</span>   <span class="dv">6</span>h<span class="fl">.10</span>nM.E2    <span class="dv">0</span>                <span class="cn">NA</span></span>
<span id="cb52-24"><a href="#cb52-24" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">3</span> unresponsive <span class="dv">8557</span>       <span class="fl">0.004011163</span></span>
<span id="cb52-25"><a href="#cb52-25" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">4</span>         null <span class="dv">4682</span>       <span class="fl">0.150392374</span></span>
<span id="cb52-26"><a href="#cb52-26" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">5</span>          FBS  <span class="dv">853</span>      <span class="sc">-</span><span class="fl">0.834965843</span></span>
<span id="cb52-27"><a href="#cb52-27" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">6</span>   <span class="dv">6</span>h<span class="fl">.10</span>nM.E2  <span class="dv">748</span>       <span class="fl">0.930139628</span></span>
<span id="cb52-28"><a href="#cb52-28" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">7</span> unresponsive <span class="dv">2358</span>      <span class="sc">-</span><span class="fl">0.010138355</span></span>
<span id="cb52-29"><a href="#cb52-29" tabindex="-1"></a><span class="sc">&gt;</span> <span class="dv">8</span>         null <span class="dv">9280</span>      <span class="sc">-</span><span class="fl">0.180716097</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level3">
<h3 id="enrichments">GSEA and OverRepresentation Analyses (ORA)<a class="anchor" aria-label="anchor" href="#enrichments"></a>
</h3>
<p>Enrichment analyses can be performed on a full contrast of a
<code>DEprot.analyses</code> object (GSEA) or on the subset of proteins
that belong to a specific category of <code>diff.status</code> in the
results table of the given contrast (ORA).</p>
<p>For these analyses a geneSet table will be required as “reference”.
The latter must be a data.frame with two columns: “gs_name” (geneSet
name) and “gene_symbol” (corresponding to the protein id).<br> In
<code>DEprot</code> we provide a ready-to-use list of protein complexes
obtained by the <a href="https://mips.helmholtz-muenchen.de/corum/" class="external-link">CORUM</a> database.</p>
<blockquote>
<p>Notice that the “gene_symbol” values must match the “prot.id” in the
result tables. To this goal, the <code>geneset.enrichment</code>
function contains the parameter <code>gsub.pattern.prot.id</code> that
will be used to remove a pattern from the “prot.id”s.</p>
</blockquote>
<p><br></p>
<p>In the following example we will show how to create a geneSet table
from the CORUM data and use it for GSEA and ORA analyses.</p>
<div class="section level4">
<h4 id="define-geneset">Define geneSet<a class="anchor" aria-label="anchor" href="#define-geneset"></a>
</h4>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## GeneSet Enrichment Analyses</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"corum_v5.0"</span>, package <span class="op">=</span> <span class="st">"DEprot"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">corum_geneSet</span> <span class="op">=</span></span>
<span>  <span class="va">corum_v5.0</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">organism</span> <span class="op">==</span> <span class="st">"Human"</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>gs_name <span class="op">=</span> <span class="va">complex.name</span>,</span>
<span>                gene_symbol <span class="op">=</span> <span class="va">protein.members</span><span class="op">)</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gs_name</span>, <span class="va">gene_symbol</span><span class="op">)</span></span>
<span></span>
<span><span class="va">corum_geneSet</span></span></code></pre></div>
<table class="table">
<caption><strong>CORUM protein complexes (v5.0)</strong></caption>
<thead><tr class="header">
<th align="left">gs_name</th>
<th align="left">gene_symbol</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">BCL6-HDAC4 complex</td>
<td align="left">BCL6</td>
</tr>
<tr class="even">
<td align="left">BCL6-HDAC4 complex</td>
<td align="left">HDAC4</td>
</tr>
<tr class="odd">
<td align="left">BCL6-HDAC5 complex</td>
<td align="left">BCL6</td>
</tr>
<tr class="even">
<td align="left">BCL6-HDAC5 complex</td>
<td align="left">HDAC5</td>
</tr>
<tr class="odd">
<td align="left">BCL6-HDAC7 complex</td>
<td align="left">BCL6</td>
</tr>
<tr class="even">
<td align="left">BCL6-HDAC7 complex</td>
<td align="left">HDAC7</td>
</tr>
<tr class="odd">
<td align="left">Multisubunit ACTR coactivator complex</td>
<td align="left">CREBBP</td>
</tr>
<tr class="even">
<td align="left">Multisubunit ACTR coactivator complex</td>
<td align="left">EP300</td>
</tr>
<tr class="odd">
<td align="left">Multisubunit ACTR coactivator complex</td>
<td align="left">KAT2B</td>
</tr>
<tr class="even">
<td align="left">Multisubunit ACTR coactivator complex</td>
<td align="left">NCOA3</td>
</tr>
</tbody>
</table>
<p><br></p>
</div>
<div class="section level4">
<h4 id="perform-enrichment-analyses">Perform enrichment analyses<a class="anchor" aria-label="anchor" href="#perform-enrichment-analyses"></a>
</h4>
<div class="section level5">
<h5 id="compare-ranking-methods-for-gsea">Compare ranking methods for GSEA<a class="anchor" aria-label="anchor" href="#compare-ranking-methods-for-gsea"></a>
</h5>
<p>For GSEA analyses the genes in the differential table of the provided
contrast can be ranked either by using the log<sub>2</sub>(Fold Change)
[default] or the Spearman’s correlation of the imputed counts between
the two groups. Depending on the sample size, this two methods can lead
to different ranking. To compare the two methods it is possible to plot
the correlation of the two ranks and scores using
<code>compare.ranking</code>:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/compare.ranking.html">compare.ranking</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                contrast <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="DEprot.overview.vignette_files/figure-html/compare_ranking-1.png" width="960" style="display: block; margin: auto;"></p>
<p><br></p>
</div>
<div class="section level5">
<h5 id="geneset-enrichment-analyses-gsea">GeneSet Enrichment Analyses (GSEA)<a class="anchor" aria-label="anchor" href="#geneset-enrichment-analyses-gsea"></a>
</h5>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">GSEA.results</span> <span class="op">=</span></span>
<span>  <span class="fu"><a href="../reference/geneset.enrichment.html">geneset.enrichment</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                     contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                     TERM2GENE <span class="op">=</span> <span class="va">corum_geneSet</span>,</span>
<span>                     enrichment.type <span class="op">=</span> <span class="st">"GSEA"</span>,</span>
<span>                     gsea.rank.method <span class="op">=</span> <span class="st">"foldchange"</span>, <span class="co"># or correlation</span></span>
<span>                     gsub.pattern.prot.id <span class="op">=</span> <span class="st">"_HUMAN|;.*"</span>,</span>
<span>                     pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                     dotplot.n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level5">
<h5 id="overrepresentation-analyses-ora">OverRepresentation Analyses (ORA)<a class="anchor" aria-label="anchor" href="#overrepresentation-analyses-ora"></a>
</h5>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ORA.results</span> <span class="op">=</span></span>
<span>  <span class="fu"><a href="../reference/geneset.enrichment.html">geneset.enrichment</a></span><span class="op">(</span>DEprot.analyses.object <span class="op">=</span> <span class="va">dpo_analyses</span>,</span>
<span>                     contrast <span class="op">=</span> <span class="fl">1</span>,</span>
<span>                     TERM2GENE <span class="op">=</span> <span class="va">corum_geneSet</span>,</span>
<span>                     enrichment.type <span class="op">=</span> <span class="st">"ORA"</span>,</span>
<span>                     gsub.pattern.prot.id <span class="op">=</span> <span class="st">"_HUMAN|;.*"</span>,</span>
<span>                     pvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     qvalueCutoff <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>                     pAdjustMethod <span class="op">=</span> <span class="st">"BH"</span>,</span>
<span>                     diff.status.category <span class="op">=</span> <span class="st">"6h.10nM.E2"</span>,</span>
<span>                     dotplot.n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p><br></p>
</div>
<div class="section level5">
<h5 id="enrichment-output">Enrichment output<a class="anchor" aria-label="anchor" href="#enrichment-output"></a>
</h5>
<p>The output is an object of class
<code>DEprot.enrichResult</code>:</p>
<table class="table">
<colgroup>
<col width="27%">
<col width="72%">
</colgroup>
<thead><tr class="header">
<th align="right">Slot</th>
<th align="left">Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="right"><em>enrichment.discovery</em></td>
<td align="left">the direct output from
<code>clusterProfiler::GSEA/enricher</code> (GSEA and ORA,
respectively)</td>
</tr>
<tr class="even">
<td align="right"><em>protein.network</em></td>
<td align="left">a string plot showing protein networks
(<code><a href="https://rdrr.io/pkg/ggtangle/man/cnetplot.html" class="external-link">clusterProfiler::cnetplot</a></code>)</td>
</tr>
<tr class="odd">
<td align="right"><em>pathway.network</em></td>
<td align="left">a list with clusters and string plot showing
pathway/set networks (<code>aPEAR::enrichmentNetwork</code>)</td>
</tr>
<tr class="even">
<td align="right">
<em>NES.plot</em> (GSEA only)</td>
<td align="left">a bar plot showing the NES scores for each
significantly enriched geneSet</td>
</tr>
<tr class="odd">
<td align="right"><em>dotplot_gene.ratio</em></td>
<td align="left">a dotplot showing the geneRatios for each significantly
enriched geneSet</td>
</tr>
<tr class="even">
<td align="right">
<em>dotplot_fold.enrichment</em> (ORA only)</td>
<td align="left">a dotplot showing the foldEnrichment for each
significantly enriched geneSet</td>
</tr>
<tr class="odd">
<td align="right"><em>parameters</em></td>
<td align="left">a list containing the parameters used to run the
analyses</td>
</tr>
</tbody>
</table>
<p><br></p>
<hr>
</div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="info">
<strong>Package information</strong><a class="anchor" aria-label="anchor" href="#info"></a>
</h2>
<div class="section level3">
<h3 id="documentation">Documentation<a class="anchor" aria-label="anchor" href="#documentation"></a>
</h3>
<p>With the package a detailed <a href="https://sebastian-gregoricchio.github.io/DEprot/DEprot_manual.pdf" class="external-link">PDF
manual</a> with details for each function and respective parameters is
available.</p>
<p>The R-package have been published on <a href="https://github.com/sebastian-gregoricchio/DEprot" class="external-link">GitHub</a> and a
<a href="https://sebastian-gregoricchio.github.io/DEprot" class="external-link">git-pages
website</a> is available as well. At both these sites it is possible to
find the installation procedure, required dependencies, and the links
for changeLog, manual and vignette.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="package-history-and-releases">Package history and releases<a class="anchor" aria-label="anchor" href="#package-history-and-releases"></a>
</h3>
<p>A list of all releases and respective description of changes applied
could be found <a href="https://sebastian-gregoricchio.github.io/DEprot/NEWS" class="external-link">here</a>.</p>
<p><br></p>
</div>
<div class="section level3">
<h3 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h3>
<p>For any suggestion, bug fixing, commentary please contact fill an <a href="https://github.com/sebastian-gregoricchio/DEprot/issues" class="external-link">Issues</a>/<a href="https://github.com/sebastian-gregoricchio/DEprot/pulls" class="external-link">Pull
requests</a> form on the <code>DEprot</code> github page.</p>
<p><img src="https://badges.pufler.dev/contributors/sebastian-gregoricchio/DEprot?size=50&amp;padding=5&amp;bots=true"></p>
</div>
<div class="section level3">
<h3 id="license">License<a class="anchor" aria-label="anchor" href="#license"></a>
</h3>
<p>This package is under a <a href="https://sebastian-gregoricchio.github.io/DEprot/LICENSE.md/LICENSE" class="external-link">GNU
General Public License (version 3)</a>.</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Sebastian Gregoricchio.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

      </footer>
</body>
</html>
